// @alicloud/dingtalk/dist/trip_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class GetTravelProcessDetailHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetTravelProcessDetailHeaders = GetTravelProcessDetailHeaders;
class GetTravelProcessDetailRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            processInstanceId: 'processInstanceId',
        };
    }
    static types() {
        return {
            processInstanceId: 'string',
        };
    }
}
exports.GetTravelProcessDetailRequest = GetTravelProcessDetailRequest;
class GetTravelProcessDetailResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: GetTravelProcessDetailResponseBodyResult,
            success: 'boolean',
        };
    }
}
exports.GetTravelProcessDetailResponseBody = GetTravelProcessDetailResponseBody;
class GetTravelProcessDetailResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetTravelProcessDetailResponseBody,
        };
    }
}
exports.GetTravelProcessDetailResponse = GetTravelProcessDetailResponse;
class SyncBusinessSignInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SyncBusinessSignInfoHeaders = SyncBusinessSignInfoHeaders;
class SyncBusinessSignInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizTypeList: 'bizTypeList',
            gmtOrgPay: 'gmtOrgPay',
            gmtSign: 'gmtSign',
            orgPayStatus: 'orgPayStatus',
            signStatus: 'signStatus',
            targetCorpId: 'targetCorpId',
            tmcProductDetailList: 'tmcProductDetailList',
            tmcProductList: 'tmcProductList',
        };
    }
    static types() {
        return {
            bizTypeList: { 'type': 'array', 'itemType': 'string' },
            gmtOrgPay: 'string',
            gmtSign: 'string',
            orgPayStatus: 'string',
            signStatus: 'string',
            targetCorpId: 'string',
            tmcProductDetailList: { 'type': 'array', 'itemType': SyncBusinessSignInfoRequestTmcProductDetailList },
            tmcProductList: { 'type': 'array', 'itemType': SyncBusinessSignInfoRequestTmcProductList },
        };
    }
}
exports.SyncBusinessSignInfoRequest = SyncBusinessSignInfoRequest;
class SyncBusinessSignInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            success: 'success',
        };
    }
    static types() {
        return {
            requestId: 'string',
            success: 'boolean',
        };
    }
}
exports.SyncBusinessSignInfoResponseBody = SyncBusinessSignInfoResponseBody;
class SyncBusinessSignInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SyncBusinessSignInfoResponseBody,
        };
    }
}
exports.SyncBusinessSignInfoResponse = SyncBusinessSignInfoResponse;
class SyncSecretKeyHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SyncSecretKeyHeaders = SyncSecretKeyHeaders;
class SyncSecretKeyRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actionType: 'actionType',
            secretString: 'secretString',
            targetCorpId: 'targetCorpId',
            tripAppKey: 'tripAppKey',
            tripAppSecurity: 'tripAppSecurity',
            tripCorpId: 'tripCorpId',
        };
    }
    static types() {
        return {
            actionType: 'string',
            secretString: 'string',
            targetCorpId: 'string',
            tripAppKey: 'string',
            tripAppSecurity: 'string',
            tripCorpId: 'string',
        };
    }
}
exports.SyncSecretKeyRequest = SyncSecretKeyRequest;
class SyncSecretKeyResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: SyncSecretKeyResponseBodyResult,
            success: 'string',
        };
    }
}
exports.SyncSecretKeyResponseBody = SyncSecretKeyResponseBody;
class SyncSecretKeyResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SyncSecretKeyResponseBody,
        };
    }
}
exports.SyncSecretKeyResponse = SyncSecretKeyResponse;
class SyncTripOrderHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SyncTripOrderHeaders = SyncTripOrderHeaders;
class SyncTripOrderRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            channelType: 'channelType',
            currency: 'currency',
            dingUserId: 'dingUserId',
            discountAmount: 'discountAmount',
            endorseFlag: 'endorseFlag',
            event: 'event',
            gmtOrder: 'gmtOrder',
            gmtPay: 'gmtPay',
            gmtRefund: 'gmtRefund',
            invoiceApplyUrl: 'invoiceApplyUrl',
            journeyBizNo: 'journeyBizNo',
            orderDetails: 'orderDetails',
            orderNo: 'orderNo',
            orderUrl: 'orderUrl',
            processId: 'processId',
            realAmount: 'realAmount',
            refundAmount: 'refundAmount',
            relativeOrderNo: 'relativeOrderNo',
            source: 'source',
            supplyLogo: 'supplyLogo',
            supplyName: 'supplyName',
            targetCorpId: 'targetCorpId',
            tmcCorpId: 'tmcCorpId',
            totalAmount: 'totalAmount',
            type: 'type',
        };
    }
    static types() {
        return {
            channelType: 'string',
            currency: 'string',
            dingUserId: 'string',
            discountAmount: 'string',
            endorseFlag: 'boolean',
            event: SyncTripOrderRequestEvent,
            gmtOrder: 'string',
            gmtPay: 'string',
            gmtRefund: 'string',
            invoiceApplyUrl: 'string',
            journeyBizNo: 'string',
            orderDetails: { 'type': 'array', 'itemType': SyncTripOrderRequestOrderDetails },
            orderNo: 'string',
            orderUrl: 'string',
            processId: 'string',
            realAmount: 'string',
            refundAmount: 'string',
            relativeOrderNo: 'string',
            source: 'string',
            supplyLogo: 'string',
            supplyName: 'string',
            targetCorpId: 'string',
            tmcCorpId: 'string',
            totalAmount: 'string',
            type: 'string',
        };
    }
}
exports.SyncTripOrderRequest = SyncTripOrderRequest;
class SyncTripOrderResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            success: 'success',
        };
    }
    static types() {
        return {
            requestId: 'string',
            success: 'boolean',
        };
    }
}
exports.SyncTripOrderResponseBody = SyncTripOrderResponseBody;
class SyncTripOrderResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SyncTripOrderResponseBody,
        };
    }
}
exports.SyncTripOrderResponse = SyncTripOrderResponse;
class GetTravelProcessDetailResponseBodyResultJourneysArrival extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            name: 'name',
            nationalCityCode: 'nationalCityCode',
        };
    }
    static types() {
        return {
            code: 'string',
            name: 'string',
            nationalCityCode: 'string',
        };
    }
}
exports.GetTravelProcessDetailResponseBodyResultJourneysArrival = GetTravelProcessDetailResponseBodyResultJourneysArrival;
class GetTravelProcessDetailResponseBodyResultJourneysDeparture extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            name: 'name',
            nationalCityCode: 'nationalCityCode',
        };
    }
    static types() {
        return {
            code: 'string',
            name: 'string',
            nationalCityCode: 'string',
        };
    }
}
exports.GetTravelProcessDetailResponseBodyResultJourneysDeparture = GetTravelProcessDetailResponseBodyResultJourneysDeparture;
class GetTravelProcessDetailResponseBodyResultJourneys extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            arrival: 'arrival',
            departure: 'departure',
            endTime: 'endTime',
            journeyBizNo: 'journeyBizNo',
            startTime: 'startTime',
            travelType: 'travelType',
            tripWay: 'tripWay',
        };
    }
    static types() {
        return {
            arrival: GetTravelProcessDetailResponseBodyResultJourneysArrival,
            departure: GetTravelProcessDetailResponseBodyResultJourneysDeparture,
            endTime: 'string',
            journeyBizNo: 'string',
            startTime: 'string',
            travelType: 'string',
            tripWay: 'string',
        };
    }
}
exports.GetTravelProcessDetailResponseBodyResultJourneys = GetTravelProcessDetailResponseBodyResultJourneys;
class GetTravelProcessDetailResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            businessId: 'businessId',
            corpId: 'corpId',
            costCenter: 'costCenter',
            itineraryProject: 'itineraryProject',
            journeys: 'journeys',
            mainProcessInstanceId: 'mainProcessInstanceId',
            memo: 'memo',
            originatorId: 'originatorId',
            processInstanceId: 'processInstanceId',
            processResult: 'processResult',
            processStatus: 'processStatus',
            remark: 'remark',
            travelCategory: 'travelCategory',
            travelers: 'travelers',
        };
    }
    static types() {
        return {
            businessId: 'string',
            corpId: 'string',
            costCenter: 'string',
            itineraryProject: 'string',
            journeys: { 'type': 'array', 'itemType': GetTravelProcessDetailResponseBodyResultJourneys },
            mainProcessInstanceId: 'string',
            memo: 'string',
            originatorId: 'string',
            processInstanceId: 'string',
            processResult: 'string',
            processStatus: 'string',
            remark: 'string',
            travelCategory: 'string',
            travelers: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetTravelProcessDetailResponseBodyResult = GetTravelProcessDetailResponseBodyResult;
class SyncBusinessSignInfoRequestTmcProductDetailList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            gmtOrgPay: 'gmtOrgPay',
            payType: 'payType',
            product: 'product',
        };
    }
    static types() {
        return {
            gmtOrgPay: 'string',
            payType: 'string',
            product: 'string',
        };
    }
}
exports.SyncBusinessSignInfoRequestTmcProductDetailList = SyncBusinessSignInfoRequestTmcProductDetailList;
class SyncBusinessSignInfoRequestTmcProductListProductDetailList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            categoryType: 'categoryType',
            gmtOrgPay: 'gmtOrgPay',
            openStatus: 'openStatus',
            payType: 'payType',
            product: 'product',
        };
    }
    static types() {
        return {
            categoryType: 'string',
            gmtOrgPay: 'string',
            openStatus: 'boolean',
            payType: 'string',
            product: 'string',
        };
    }
}
exports.SyncBusinessSignInfoRequestTmcProductListProductDetailList = SyncBusinessSignInfoRequestTmcProductListProductDetailList;
class SyncBusinessSignInfoRequestTmcProductList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            productDetailList: 'productDetailList',
            tmcCorpId: 'tmcCorpId',
        };
    }
    static types() {
        return {
            productDetailList: { 'type': 'array', 'itemType': SyncBusinessSignInfoRequestTmcProductListProductDetailList },
            tmcCorpId: 'string',
        };
    }
}
exports.SyncBusinessSignInfoRequestTmcProductList = SyncBusinessSignInfoRequestTmcProductList;
class SyncSecretKeyResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            secretString: 'secretString',
            targetCorpId: 'targetCorpId',
            tripAppKey: 'tripAppKey',
            tripAppSecurity: 'tripAppSecurity',
            tripCorpId: 'tripCorpId',
        };
    }
    static types() {
        return {
            secretString: 'string',
            targetCorpId: 'string',
            tripAppKey: 'string',
            tripAppSecurity: 'string',
            tripCorpId: 'string',
        };
    }
}
exports.SyncSecretKeyResponseBodyResult = SyncSecretKeyResponseBodyResult;
class SyncTripOrderRequestEvent extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            action: 'action',
            gmtAction: 'gmtAction',
        };
    }
    static types() {
        return {
            action: 'string',
            gmtAction: 'string',
        };
    }
}
exports.SyncTripOrderRequestEvent = SyncTripOrderRequestEvent;
class SyncTripOrderRequestOrderDetailsHotelLocation extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            lat: 'lat',
            lon: 'lon',
            source: 'source',
            url: 'url',
        };
    }
    static types() {
        return {
            lat: 'string',
            lon: 'string',
            source: 'string',
            url: 'string',
        };
    }
}
exports.SyncTripOrderRequestOrderDetailsHotelLocation = SyncTripOrderRequestOrderDetailsHotelLocation;
class SyncTripOrderRequestOrderDetailsOpenConsumerInfo extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            name: 'name',
            staffFlag: 'staffFlag',
            status: 'status',
            ticketAmount: 'ticketAmount',
            ticketNo: 'ticketNo',
            userId: 'userId',
        };
    }
    static types() {
        return {
            corpId: 'string',
            name: 'string',
            staffFlag: 'boolean',
            status: 'string',
            ticketAmount: 'string',
            ticketNo: 'string',
            userId: 'string',
        };
    }
}
exports.SyncTripOrderRequestOrderDetailsOpenConsumerInfo = SyncTripOrderRequestOrderDetailsOpenConsumerInfo;
class SyncTripOrderRequestOrderDetails extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            arrivalTime: 'arrivalTime',
            carColor: 'carColor',
            carModel: 'carModel',
            carNumber: 'carNumber',
            cateringType: 'cateringType',
            checkInTime: 'checkInTime',
            checkOutTime: 'checkOutTime',
            departTime: 'departTime',
            destinationCity: 'destinationCity',
            destinationCityCode: 'destinationCityCode',
            destinationStation: 'destinationStation',
            detailAmount: 'detailAmount',
            hotelAddress: 'hotelAddress',
            hotelCity: 'hotelCity',
            hotelLocation: 'hotelLocation',
            hotelName: 'hotelName',
            openConsumerInfo: 'openConsumerInfo',
            originCity: 'originCity',
            originCityCode: 'originCityCode',
            originStation: 'originStation',
            roomCount: 'roomCount',
            seatInfo: 'seatInfo',
            serviceType: 'serviceType',
            subSupplyLogo: 'subSupplyLogo',
            subSupplyName: 'subSupplyName',
            taxiType: 'taxiType',
            telephone: 'telephone',
            transportNumber: 'transportNumber',
            typeDescription: 'typeDescription',
        };
    }
    static types() {
        return {
            arrivalTime: 'string',
            carColor: 'string',
            carModel: 'string',
            carNumber: 'string',
            cateringType: 'string',
            checkInTime: 'string',
            checkOutTime: 'string',
            departTime: 'string',
            destinationCity: 'string',
            destinationCityCode: 'string',
            destinationStation: 'string',
            detailAmount: 'string',
            hotelAddress: 'string',
            hotelCity: 'string',
            hotelLocation: SyncTripOrderRequestOrderDetailsHotelLocation,
            hotelName: 'string',
            openConsumerInfo: { 'type': 'array', 'itemType': SyncTripOrderRequestOrderDetailsOpenConsumerInfo },
            originCity: 'string',
            originCityCode: 'string',
            originStation: 'string',
            roomCount: 'number',
            seatInfo: 'string',
            serviceType: 'string',
            subSupplyLogo: 'string',
            subSupplyName: 'string',
            taxiType: 'string',
            telephone: 'string',
            transportNumber: 'string',
            typeDescription: 'string',
        };
    }
}
exports.SyncTripOrderRequestOrderDetails = SyncTripOrderRequestOrderDetails;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async getTravelProcessDetailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.processInstanceId)) {
            query["processInstanceId"] = request.processInstanceId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetTravelProcessDetail",
            version: "trip_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/trip/processes/details`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetTravelProcessDetailResponse({}));
    }
    async getTravelProcessDetail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetTravelProcessDetailHeaders({});
        return await this.getTravelProcessDetailWithOptions(request, headers, runtime);
    }
    async syncBusinessSignInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizTypeList)) {
            body["bizTypeList"] = request.bizTypeList;
        }
        if (!tea_util_1.default.isUnset(request.gmtOrgPay)) {
            body["gmtOrgPay"] = request.gmtOrgPay;
        }
        if (!tea_util_1.default.isUnset(request.gmtSign)) {
            body["gmtSign"] = request.gmtSign;
        }
        if (!tea_util_1.default.isUnset(request.orgPayStatus)) {
            body["orgPayStatus"] = request.orgPayStatus;
        }
        if (!tea_util_1.default.isUnset(request.signStatus)) {
            body["signStatus"] = request.signStatus;
        }
        if (!tea_util_1.default.isUnset(request.targetCorpId)) {
            body["targetCorpId"] = request.targetCorpId;
        }
        if (!tea_util_1.default.isUnset(request.tmcProductDetailList)) {
            body["tmcProductDetailList"] = request.tmcProductDetailList;
        }
        if (!tea_util_1.default.isUnset(request.tmcProductList)) {
            body["tmcProductList"] = request.tmcProductList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SyncBusinessSignInfo",
            version: "trip_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/trip/businessSignInfos/sync`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SyncBusinessSignInfoResponse({}));
    }
    async syncBusinessSignInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SyncBusinessSignInfoHeaders({});
        return await this.syncBusinessSignInfoWithOptions(request, headers, runtime);
    }
    async syncSecretKeyWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.actionType)) {
            body["actionType"] = request.actionType;
        }
        if (!tea_util_1.default.isUnset(request.secretString)) {
            body["secretString"] = request.secretString;
        }
        if (!tea_util_1.default.isUnset(request.targetCorpId)) {
            body["targetCorpId"] = request.targetCorpId;
        }
        if (!tea_util_1.default.isUnset(request.tripAppKey)) {
            body["tripAppKey"] = request.tripAppKey;
        }
        if (!tea_util_1.default.isUnset(request.tripAppSecurity)) {
            body["tripAppSecurity"] = request.tripAppSecurity;
        }
        if (!tea_util_1.default.isUnset(request.tripCorpId)) {
            body["tripCorpId"] = request.tripCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SyncSecretKey",
            version: "trip_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/trip/secretKeys/sync`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SyncSecretKeyResponse({}));
    }
    async syncSecretKey(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SyncSecretKeyHeaders({});
        return await this.syncSecretKeyWithOptions(request, headers, runtime);
    }
    async syncTripOrderWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.channelType)) {
            body["channelType"] = request.channelType;
        }
        if (!tea_util_1.default.isUnset(request.currency)) {
            body["currency"] = request.currency;
        }
        if (!tea_util_1.default.isUnset(request.dingUserId)) {
            body["dingUserId"] = request.dingUserId;
        }
        if (!tea_util_1.default.isUnset(request.discountAmount)) {
            body["discountAmount"] = request.discountAmount;
        }
        if (!tea_util_1.default.isUnset(request.endorseFlag)) {
            body["endorseFlag"] = request.endorseFlag;
        }
        if (!tea_util_1.default.isUnset(request.event)) {
            body["event"] = request.event;
        }
        if (!tea_util_1.default.isUnset(request.gmtOrder)) {
            body["gmtOrder"] = request.gmtOrder;
        }
        if (!tea_util_1.default.isUnset(request.gmtPay)) {
            body["gmtPay"] = request.gmtPay;
        }
        if (!tea_util_1.default.isUnset(request.gmtRefund)) {
            body["gmtRefund"] = request.gmtRefund;
        }
        if (!tea_util_1.default.isUnset(request.invoiceApplyUrl)) {
            body["invoiceApplyUrl"] = request.invoiceApplyUrl;
        }
        if (!tea_util_1.default.isUnset(request.journeyBizNo)) {
            body["journeyBizNo"] = request.journeyBizNo;
        }
        if (!tea_util_1.default.isUnset(request.orderDetails)) {
            body["orderDetails"] = request.orderDetails;
        }
        if (!tea_util_1.default.isUnset(request.orderNo)) {
            body["orderNo"] = request.orderNo;
        }
        if (!tea_util_1.default.isUnset(request.orderUrl)) {
            body["orderUrl"] = request.orderUrl;
        }
        if (!tea_util_1.default.isUnset(request.processId)) {
            body["processId"] = request.processId;
        }
        if (!tea_util_1.default.isUnset(request.realAmount)) {
            body["realAmount"] = request.realAmount;
        }
        if (!tea_util_1.default.isUnset(request.refundAmount)) {
            body["refundAmount"] = request.refundAmount;
        }
        if (!tea_util_1.default.isUnset(request.relativeOrderNo)) {
            body["relativeOrderNo"] = request.relativeOrderNo;
        }
        if (!tea_util_1.default.isUnset(request.source)) {
            body["source"] = request.source;
        }
        if (!tea_util_1.default.isUnset(request.supplyLogo)) {
            body["supplyLogo"] = request.supplyLogo;
        }
        if (!tea_util_1.default.isUnset(request.supplyName)) {
            body["supplyName"] = request.supplyName;
        }
        if (!tea_util_1.default.isUnset(request.targetCorpId)) {
            body["targetCorpId"] = request.targetCorpId;
        }
        if (!tea_util_1.default.isUnset(request.tmcCorpId)) {
            body["tmcCorpId"] = request.tmcCorpId;
        }
        if (!tea_util_1.default.isUnset(request.totalAmount)) {
            body["totalAmount"] = request.totalAmount;
        }
        if (!tea_util_1.default.isUnset(request.type)) {
            body["type"] = request.type;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SyncTripOrder",
            version: "trip_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/trip/tripOrders/sync`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SyncTripOrderResponse({}));
    }
    async syncTripOrder(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SyncTripOrderHeaders({});
        return await this.syncTripOrderWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
