// @alicloud/dingtalk/dist/smart_device_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddDeviceVideoConferenceMembersHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AddDeviceVideoConferenceMembersHeaders = AddDeviceVideoConferenceMembersHeaders;
class AddDeviceVideoConferenceMembersRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.AddDeviceVideoConferenceMembersRequest = AddDeviceVideoConferenceMembersRequest;
class AddDeviceVideoConferenceMembersResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
}
exports.AddDeviceVideoConferenceMembersResponse = AddDeviceVideoConferenceMembersResponse;
class CreateDeviceVideoConferenceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateDeviceVideoConferenceHeaders = CreateDeviceVideoConferenceHeaders;
class CreateDeviceVideoConferenceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.CreateDeviceVideoConferenceRequest = CreateDeviceVideoConferenceRequest;
class CreateDeviceVideoConferenceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            conferenceId: 'conferenceId',
        };
    }
    static types() {
        return {
            code: 'string',
            conferenceId: 'string',
        };
    }
}
exports.CreateDeviceVideoConferenceResponseBody = CreateDeviceVideoConferenceResponseBody;
class CreateDeviceVideoConferenceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateDeviceVideoConferenceResponseBody,
        };
    }
}
exports.CreateDeviceVideoConferenceResponse = CreateDeviceVideoConferenceResponse;
class ExtractFacialFeatureHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ExtractFacialFeatureHeaders = ExtractFacialFeatureHeaders;
class ExtractFacialFeatureRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mediaId: 'mediaId',
            userid: 'userid',
        };
    }
    static types() {
        return {
            mediaId: 'string',
            userid: 'string',
        };
    }
}
exports.ExtractFacialFeatureRequest = ExtractFacialFeatureRequest;
class ExtractFacialFeatureResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.ExtractFacialFeatureResponseBody = ExtractFacialFeatureResponseBody;
class ExtractFacialFeatureResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ExtractFacialFeatureResponseBody,
        };
    }
}
exports.ExtractFacialFeatureResponse = ExtractFacialFeatureResponse;
class KickDeviceVideoConferenceMembersHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.KickDeviceVideoConferenceMembersHeaders = KickDeviceVideoConferenceMembersHeaders;
class KickDeviceVideoConferenceMembersRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.KickDeviceVideoConferenceMembersRequest = KickDeviceVideoConferenceMembersRequest;
class KickDeviceVideoConferenceMembersResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
}
exports.KickDeviceVideoConferenceMembersResponse = KickDeviceVideoConferenceMembersResponse;
class MachineManagerUpdateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.MachineManagerUpdateHeaders = MachineManagerUpdateHeaders;
class MachineManagerUpdateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            atmManagerRightMap: 'atmManagerRightMap',
            deviceId: 'deviceId',
            scopeDeptIds: 'scopeDeptIds',
            userId: 'userId',
        };
    }
    static types() {
        return {
            atmManagerRightMap: MachineManagerUpdateRequestAtmManagerRightMap,
            deviceId: 'number',
            scopeDeptIds: { 'type': 'array', 'itemType': 'number' },
            userId: 'string',
        };
    }
}
exports.MachineManagerUpdateRequest = MachineManagerUpdateRequest;
class MachineManagerUpdateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
}
exports.MachineManagerUpdateResponse = MachineManagerUpdateResponse;
class MachineUsersUpdateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.MachineUsersUpdateHeaders = MachineUsersUpdateHeaders;
class MachineUsersUpdateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            addDeptIds: 'addDeptIds',
            addUserIds: 'addUserIds',
            delDeptIds: 'delDeptIds',
            delUserIds: 'delUserIds',
            devIds: 'devIds',
            deviceIds: 'deviceIds',
        };
    }
    static types() {
        return {
            addDeptIds: { 'type': 'array', 'itemType': 'number' },
            addUserIds: { 'type': 'array', 'itemType': 'string' },
            delDeptIds: { 'type': 'array', 'itemType': 'number' },
            delUserIds: { 'type': 'array', 'itemType': 'string' },
            devIds: { 'type': 'array', 'itemType': 'number' },
            deviceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.MachineUsersUpdateRequest = MachineUsersUpdateRequest;
class MachineUsersUpdateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
        };
    }
}
exports.MachineUsersUpdateResponse = MachineUsersUpdateResponse;
class QueryDeviceVideoConferenceBookHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryDeviceVideoConferenceBookHeaders = QueryDeviceVideoConferenceBookHeaders;
class QueryDeviceVideoConferenceBookResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            conferenceId: 'conferenceId',
        };
    }
    static types() {
        return {
            code: 'string',
            conferenceId: 'string',
        };
    }
}
exports.QueryDeviceVideoConferenceBookResponseBody = QueryDeviceVideoConferenceBookResponseBody;
class QueryDeviceVideoConferenceBookResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryDeviceVideoConferenceBookResponseBody,
        };
    }
}
exports.QueryDeviceVideoConferenceBookResponse = QueryDeviceVideoConferenceBookResponse;
class MachineManagerUpdateRequestAtmManagerRightMap extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            attendancePersonManage: 'attendancePersonManage',
            bluetoothPunchManage: 'bluetoothPunchManage',
            deviceReset: 'deviceReset',
            deviceSettings: 'deviceSettings',
            facePunchManage: 'facePunchManage',
            fingerPunchManage: 'fingerPunchManage',
        };
    }
    static types() {
        return {
            attendancePersonManage: 'boolean',
            bluetoothPunchManage: 'boolean',
            deviceReset: 'boolean',
            deviceSettings: 'boolean',
            facePunchManage: 'boolean',
            fingerPunchManage: 'boolean',
        };
    }
}
exports.MachineManagerUpdateRequestAtmManagerRightMap = MachineManagerUpdateRequestAtmManagerRightMap;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async addDeviceVideoConferenceMembersWithOptions(deviceId, conferenceId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddDeviceVideoConferenceMembers",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/devices/${deviceId}/videoConferences/${conferenceId}/members`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "none",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddDeviceVideoConferenceMembersResponse({}));
    }
    async addDeviceVideoConferenceMembers(deviceId, conferenceId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddDeviceVideoConferenceMembersHeaders({});
        return await this.addDeviceVideoConferenceMembersWithOptions(deviceId, conferenceId, request, headers, runtime);
    }
    async createDeviceVideoConferenceWithOptions(deviceId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateDeviceVideoConference",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/devices/${deviceId}/videoConferences`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateDeviceVideoConferenceResponse({}));
    }
    async createDeviceVideoConference(deviceId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateDeviceVideoConferenceHeaders({});
        return await this.createDeviceVideoConferenceWithOptions(deviceId, request, headers, runtime);
    }
    async extractFacialFeatureWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.mediaId)) {
            body["mediaId"] = request.mediaId;
        }
        if (!tea_util_1.default.isUnset(request.userid)) {
            body["userid"] = request.userid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ExtractFacialFeature",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/faceRecognitions/features/extract`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ExtractFacialFeatureResponse({}));
    }
    async extractFacialFeature(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ExtractFacialFeatureHeaders({});
        return await this.extractFacialFeatureWithOptions(request, headers, runtime);
    }
    async kickDeviceVideoConferenceMembersWithOptions(deviceId, conferenceId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "KickDeviceVideoConferenceMembers",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/devices/${deviceId}/videoConferences/${conferenceId}/members/batchDelete`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "none",
        });
        return $tea.cast(await this.execute(params, req, runtime), new KickDeviceVideoConferenceMembersResponse({}));
    }
    async kickDeviceVideoConferenceMembers(deviceId, conferenceId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new KickDeviceVideoConferenceMembersHeaders({});
        return await this.kickDeviceVideoConferenceMembersWithOptions(deviceId, conferenceId, request, headers, runtime);
    }
    async machineManagerUpdateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.atmManagerRightMap)) {
            body["atmManagerRightMap"] = request.atmManagerRightMap;
        }
        if (!tea_util_1.default.isUnset(request.deviceId)) {
            body["deviceId"] = request.deviceId;
        }
        if (!tea_util_1.default.isUnset(request.scopeDeptIds)) {
            body["scopeDeptIds"] = request.scopeDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "MachineManagerUpdate",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/atmachines/managers`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "none",
        });
        return $tea.cast(await this.execute(params, req, runtime), new MachineManagerUpdateResponse({}));
    }
    async machineManagerUpdate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new MachineManagerUpdateHeaders({});
        return await this.machineManagerUpdateWithOptions(request, headers, runtime);
    }
    async machineUsersUpdateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.addDeptIds)) {
            body["addDeptIds"] = request.addDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.addUserIds)) {
            body["addUserIds"] = request.addUserIds;
        }
        if (!tea_util_1.default.isUnset(request.delDeptIds)) {
            body["delDeptIds"] = request.delDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.delUserIds)) {
            body["delUserIds"] = request.delUserIds;
        }
        if (!tea_util_1.default.isUnset(request.devIds)) {
            body["devIds"] = request.devIds;
        }
        if (!tea_util_1.default.isUnset(request.deviceIds)) {
            body["deviceIds"] = request.deviceIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "MachineUsersUpdate",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/atmachines/users`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "none",
        });
        return $tea.cast(await this.execute(params, req, runtime), new MachineUsersUpdateResponse({}));
    }
    async machineUsersUpdate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new MachineUsersUpdateHeaders({});
        return await this.machineUsersUpdateWithOptions(request, headers, runtime);
    }
    async queryDeviceVideoConferenceBookWithOptions(deviceId, bookId, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "QueryDeviceVideoConferenceBook",
            version: "smartDevice_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/smartDevice/devices/${deviceId}/books/${bookId}`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryDeviceVideoConferenceBookResponse({}));
    }
    async queryDeviceVideoConferenceBook(deviceId, bookId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryDeviceVideoConferenceBookHeaders({});
        return await this.queryDeviceVideoConferenceBookWithOptions(deviceId, bookId, headers, runtime);
    }
}
exports.default = Client;
