// @alicloud/dingtalk/dist/crm_2_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class GetRelationUkSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetRelationUkSettingHeaders = GetRelationUkSettingHeaders;
class GetRelationUkSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            relationType: 'relationType',
        };
    }
    static types() {
        return {
            relationType: 'string',
        };
    }
}
exports.GetRelationUkSettingRequest = GetRelationUkSettingRequest;
class GetRelationUkSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetRelationUkSettingResponseBodyResult,
        };
    }
}
exports.GetRelationUkSettingResponseBody = GetRelationUkSettingResponseBody;
class GetRelationUkSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetRelationUkSettingResponseBody,
        };
    }
}
exports.GetRelationUkSettingResponse = GetRelationUkSettingResponse;
class GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizAlias: 'bizAlias',
            fieldId: 'fieldId',
        };
    }
    static types() {
        return {
            bizAlias: 'string',
            fieldId: 'string',
        };
    }
}
exports.GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList = GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList;
class GetRelationUkSettingResponseBodyResultFormUkSettings extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fieldList: 'fieldList',
        };
    }
    static types() {
        return {
            fieldList: { 'type': 'array', 'itemType': GetRelationUkSettingResponseBodyResultFormUkSettingsFieldList },
        };
    }
}
exports.GetRelationUkSettingResponseBodyResultFormUkSettings = GetRelationUkSettingResponseBodyResultFormUkSettings;
class GetRelationUkSettingResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            formUkSettings: 'formUkSettings',
            headerFieldIds: 'headerFieldIds',
        };
    }
    static types() {
        return {
            formUkSettings: { 'type': 'array', 'itemType': GetRelationUkSettingResponseBodyResultFormUkSettings },
            headerFieldIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetRelationUkSettingResponseBodyResult = GetRelationUkSettingResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async getRelationUkSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.relationType)) {
            query["relationType"] = request.relationType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetRelationUkSetting",
            version: "crm_2.0",
            protocol: "HTTP",
            pathname: `/v2.0/crm/relationUkSettings`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetRelationUkSettingResponse({}));
    }
    async getRelationUkSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetRelationUkSettingHeaders({});
        return await this.getRelationUkSettingWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
