// @alicloud/dingtalk/dist/package_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CloseHPackageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CloseHPackageHeaders = CloseHPackageHeaders;
class CloseHPackageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
}
exports.CloseHPackageRequest = CloseHPackageRequest;
class CloseHPackageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'any',
        };
    }
}
exports.CloseHPackageResponseBody = CloseHPackageResponseBody;
class CloseHPackageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CloseHPackageResponseBody,
        };
    }
}
exports.CloseHPackageResponse = CloseHPackageResponse;
class GetUploadTokenHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetUploadTokenHeaders = GetUploadTokenHeaders;
class GetUploadTokenRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
}
exports.GetUploadTokenRequest = GetUploadTokenRequest;
class GetUploadTokenResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accessKeyId: 'accessKeyId',
            accessKeySecret: 'accessKeySecret',
            bucket: 'bucket',
            endpoint: 'endpoint',
            expiration: 'expiration',
            name: 'name',
            region: 'region',
            stsToken: 'stsToken',
        };
    }
    static types() {
        return {
            accessKeyId: 'string',
            accessKeySecret: 'string',
            bucket: 'string',
            endpoint: 'string',
            expiration: 'string',
            name: 'string',
            region: 'string',
            stsToken: 'string',
        };
    }
}
exports.GetUploadTokenResponseBody = GetUploadTokenResponseBody;
class GetUploadTokenResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetUploadTokenResponseBody,
        };
    }
}
exports.GetUploadTokenResponse = GetUploadTokenResponse;
class HPackageListGetHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.HPackageListGetHeaders = HPackageListGetHeaders;
class HPackageListGetRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
}
exports.HPackageListGetRequest = HPackageListGetRequest;
class HPackageListGetResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            list: 'list',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            list: { 'type': 'array', 'itemType': HPackageListGetResponseBodyList },
            totalCount: 'number',
        };
    }
}
exports.HPackageListGetResponseBody = HPackageListGetResponseBody;
class HPackageListGetResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: HPackageListGetResponseBody,
        };
    }
}
exports.HPackageListGetResponse = HPackageListGetResponse;
class HPublishPackageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.HPublishPackageHeaders = HPublishPackageHeaders;
class HPublishPackageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            version: 'string',
        };
    }
}
exports.HPublishPackageRequest = HPublishPackageRequest;
class HPublishPackageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.HPublishPackageResponseBody = HPublishPackageResponseBody;
class HPublishPackageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: HPublishPackageResponseBody,
        };
    }
}
exports.HPublishPackageResponse = HPublishPackageResponse;
class HUploadPackageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.HUploadPackageHeaders = HUploadPackageHeaders;
class HUploadPackageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            ossObjectKey: 'ossObjectKey',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            ossObjectKey: 'string',
        };
    }
}
exports.HUploadPackageRequest = HUploadPackageRequest;
class HUploadPackageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            taskId: 'taskId',
        };
    }
    static types() {
        return {
            taskId: 'string',
        };
    }
}
exports.HUploadPackageResponseBody = HUploadPackageResponseBody;
class HUploadPackageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: HUploadPackageResponseBody,
        };
    }
}
exports.HUploadPackageResponse = HUploadPackageResponse;
class HUploadPackageStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.HUploadPackageStatusHeaders = HUploadPackageStatusHeaders;
class HUploadPackageStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            taskId: 'taskId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            taskId: 'string',
        };
    }
}
exports.HUploadPackageStatusRequest = HUploadPackageStatusRequest;
class HUploadPackageStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            buildTime: 'buildTime',
            finished: 'finished',
            packageSize: 'packageSize',
            status: 'status',
            taskId: 'taskId',
            version: 'version',
        };
    }
    static types() {
        return {
            buildTime: 'number',
            finished: 'boolean',
            packageSize: 'number',
            status: 'string',
            taskId: 'string',
            version: 'string',
        };
    }
}
exports.HUploadPackageStatusResponseBody = HUploadPackageStatusResponseBody;
class HUploadPackageStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: HUploadPackageStatusResponseBody,
        };
    }
}
exports.HUploadPackageStatusResponse = HUploadPackageStatusResponse;
class OpenMicroAppPackageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.OpenMicroAppPackageHeaders = OpenMicroAppPackageHeaders;
class OpenMicroAppPackageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            agentId: 'agentId',
        };
    }
    static types() {
        return {
            agentId: 'number',
        };
    }
}
exports.OpenMicroAppPackageRequest = OpenMicroAppPackageRequest;
class OpenMicroAppPackageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
}
exports.OpenMicroAppPackageResponseBody = OpenMicroAppPackageResponseBody;
class OpenMicroAppPackageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: OpenMicroAppPackageResponseBody,
        };
    }
}
exports.OpenMicroAppPackageResponse = OpenMicroAppPackageResponse;
class ReleaseGrayDeployHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayDeployHeaders = ReleaseGrayDeployHeaders;
class ReleaseGrayDeployRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            version: 'string',
        };
    }
}
exports.ReleaseGrayDeployRequest = ReleaseGrayDeployRequest;
class ReleaseGrayDeployResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'any',
        };
    }
}
exports.ReleaseGrayDeployResponseBody = ReleaseGrayDeployResponseBody;
class ReleaseGrayDeployResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayDeployResponseBody,
        };
    }
}
exports.ReleaseGrayDeployResponse = ReleaseGrayDeployResponse;
class ReleaseGrayExitHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayExitHeaders = ReleaseGrayExitHeaders;
class ReleaseGrayExitRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            version: 'string',
        };
    }
}
exports.ReleaseGrayExitRequest = ReleaseGrayExitRequest;
class ReleaseGrayExitResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            reuslt: 'reuslt',
        };
    }
    static types() {
        return {
            reuslt: 'any',
        };
    }
}
exports.ReleaseGrayExitResponseBody = ReleaseGrayExitResponseBody;
class ReleaseGrayExitResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayExitResponseBody,
        };
    }
}
exports.ReleaseGrayExitResponse = ReleaseGrayExitResponse;
class ReleaseGrayOrgGetHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayOrgGetHeaders = ReleaseGrayOrgGetHeaders;
class ReleaseGrayOrgGetRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            version: 'string',
        };
    }
}
exports.ReleaseGrayOrgGetRequest = ReleaseGrayOrgGetRequest;
class ReleaseGrayOrgGetResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            value: 'value',
        };
    }
    static types() {
        return {
            value: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.ReleaseGrayOrgGetResponseBody = ReleaseGrayOrgGetResponseBody;
class ReleaseGrayOrgGetResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayOrgGetResponseBody,
        };
    }
}
exports.ReleaseGrayOrgGetResponse = ReleaseGrayOrgGetResponse;
class ReleaseGrayOrgSetHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayOrgSetHeaders = ReleaseGrayOrgSetHeaders;
class ReleaseGrayOrgSetRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            value: 'value',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            value: { 'type': 'array', 'itemType': 'string' },
            version: 'string',
        };
    }
}
exports.ReleaseGrayOrgSetRequest = ReleaseGrayOrgSetRequest;
class ReleaseGrayOrgSetResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'any',
        };
    }
}
exports.ReleaseGrayOrgSetResponseBody = ReleaseGrayOrgSetResponseBody;
class ReleaseGrayOrgSetResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayOrgSetResponseBody,
        };
    }
}
exports.ReleaseGrayOrgSetResponse = ReleaseGrayOrgSetResponse;
class ReleaseGrayPercentGetHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayPercentGetHeaders = ReleaseGrayPercentGetHeaders;
class ReleaseGrayPercentGetRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            version: 'string',
        };
    }
}
exports.ReleaseGrayPercentGetRequest = ReleaseGrayPercentGetRequest;
class ReleaseGrayPercentGetResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            value: 'value',
        };
    }
    static types() {
        return {
            value: 'number',
        };
    }
}
exports.ReleaseGrayPercentGetResponseBody = ReleaseGrayPercentGetResponseBody;
class ReleaseGrayPercentGetResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayPercentGetResponseBody,
        };
    }
}
exports.ReleaseGrayPercentGetResponse = ReleaseGrayPercentGetResponse;
class ReleaseGrayPercentSetHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayPercentSetHeaders = ReleaseGrayPercentSetHeaders;
class ReleaseGrayPercentSetRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            value: 'value',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            value: 'number',
            version: 'string',
        };
    }
}
exports.ReleaseGrayPercentSetRequest = ReleaseGrayPercentSetRequest;
class ReleaseGrayPercentSetResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'any',
        };
    }
}
exports.ReleaseGrayPercentSetResponseBody = ReleaseGrayPercentSetResponseBody;
class ReleaseGrayPercentSetResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayPercentSetResponseBody,
        };
    }
}
exports.ReleaseGrayPercentSetResponse = ReleaseGrayPercentSetResponse;
class ReleaseGrayUserIdGetHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReleaseGrayUserIdGetHeaders = ReleaseGrayUserIdGetHeaders;
class ReleaseGrayUserIdGetRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
            version: 'version',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
            version: 'string',
        };
    }
}
exports.ReleaseGrayUserIdGetRequest = ReleaseGrayUserIdGetRequest;
class ReleaseGrayUserIdGetResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            value: 'value',
        };
    }
    static types() {
        return {
            value: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.ReleaseGrayUserIdGetResponseBody = ReleaseGrayUserIdGetResponseBody;
class ReleaseGrayUserIdGetResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReleaseGrayUserIdGetResponseBody,
        };
    }
}
exports.ReleaseGrayUserIdGetResponse = ReleaseGrayUserIdGetResponse;
class HPackageListGetResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avaliable: 'avaliable',
            creator: 'creator',
            finished: 'finished',
            operationTime: 'operationTime',
            packageSize: 'packageSize',
            status: 'status',
            taskId: 'taskId',
            version: 'version',
        };
    }
    static types() {
        return {
            avaliable: 'number',
            creator: 'string',
            finished: 'boolean',
            operationTime: 'number',
            packageSize: 'number',
            status: 'string',
            taskId: 'string',
            version: 'string',
        };
    }
}
exports.HPackageListGetResponseBodyList = HPackageListGetResponseBodyList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async closeHPackageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CloseHPackage",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/h5/microApps/close`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CloseHPackageResponse({}));
    }
    async closeHPackage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CloseHPackageHeaders({});
        return await this.closeHPackageWithOptions(request, headers, runtime);
    }
    async getUploadTokenWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetUploadToken",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/uploadTokens`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetUploadTokenResponse({}));
    }
    async getUploadToken(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUploadTokenHeaders({});
        return await this.getUploadTokenWithOptions(request, headers, runtime);
    }
    async hPackageListGetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "HPackageListGet",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/h5/versions`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new HPackageListGetResponse({}));
    }
    async hPackageListGet(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new HPackageListGetHeaders({});
        return await this.hPackageListGetWithOptions(request, headers, runtime);
    }
    async hPublishPackageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            body["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "HPublishPackage",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/h5/publish`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new HPublishPackageResponse({}));
    }
    async hPublishPackage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new HPublishPackageHeaders({});
        return await this.hPublishPackageWithOptions(request, headers, runtime);
    }
    async hUploadPackageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.ossObjectKey)) {
            body["ossObjectKey"] = request.ossObjectKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "HUploadPackage",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/h5/asyncUpload`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new HUploadPackageResponse({}));
    }
    async hUploadPackage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new HUploadPackageHeaders({});
        return await this.hUploadPackageWithOptions(request, headers, runtime);
    }
    async hUploadPackageStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.taskId)) {
            query["taskId"] = request.taskId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "HUploadPackageStatus",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/h5/uploadStatus`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new HUploadPackageStatusResponse({}));
    }
    async hUploadPackageStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new HUploadPackageStatusHeaders({});
        return await this.hUploadPackageStatusWithOptions(request, headers, runtime);
    }
    async openMicroAppPackageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.agentId)) {
            body["agentId"] = request.agentId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "OpenMicroAppPackage",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/h5/microApps/open`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new OpenMicroAppPackageResponse({}));
    }
    async openMicroAppPackage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new OpenMicroAppPackageHeaders({});
        return await this.openMicroAppPackageWithOptions(request, headers, runtime);
    }
    async releaseGrayDeployWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            body["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayDeploy",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/deploy`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayDeployResponse({}));
    }
    async releaseGrayDeploy(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayDeployHeaders({});
        return await this.releaseGrayDeployWithOptions(request, headers, runtime);
    }
    async releaseGrayExitWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            body["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayExit",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/exit`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayExitResponse({}));
    }
    async releaseGrayExit(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayExitHeaders({});
        return await this.releaseGrayExitWithOptions(request, headers, runtime);
    }
    async releaseGrayOrgGetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            query["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayOrgGet",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/organizations`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayOrgGetResponse({}));
    }
    async releaseGrayOrgGet(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayOrgGetHeaders({});
        return await this.releaseGrayOrgGetWithOptions(request, headers, runtime);
    }
    async releaseGrayOrgSetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.value)) {
            body["value"] = request.value;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            body["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayOrgSet",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/organizations/release`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayOrgSetResponse({}));
    }
    async releaseGrayOrgSet(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayOrgSetHeaders({});
        return await this.releaseGrayOrgSetWithOptions(request, headers, runtime);
    }
    async releaseGrayPercentGetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            query["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayPercentGet",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/users/percents`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayPercentGetResponse({}));
    }
    async releaseGrayPercentGet(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayPercentGetHeaders({});
        return await this.releaseGrayPercentGetWithOptions(request, headers, runtime);
    }
    async releaseGrayPercentSetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            body["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.value)) {
            body["value"] = request.value;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            body["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayPercentSet",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/users/percents/release`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayPercentSetResponse({}));
    }
    async releaseGrayPercentSet(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayPercentSetHeaders({});
        return await this.releaseGrayPercentSetWithOptions(request, headers, runtime);
    }
    async releaseGrayUserIdGetWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        if (!tea_util_1.default.isUnset(request.version)) {
            query["version"] = request.version;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ReleaseGrayUserIdGet",
            version: "package_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/package/greys/users`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReleaseGrayUserIdGetResponse({}));
    }
    async releaseGrayUserIdGet(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReleaseGrayUserIdGetHeaders({});
        return await this.releaseGrayUserIdGetWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
