// @alicloud/dingtalk/dist/link_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class DetailUserIdPrivateDataMapValue extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardParamMap: 'cardParamMap',
            cardMediaIdParamMap: 'cardMediaIdParamMap',
        };
    }
    static types() {
        return {
            cardParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            cardMediaIdParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.DetailUserIdPrivateDataMapValue = DetailUserIdPrivateDataMapValue;
class ApplyFollowerAuthInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ApplyFollowerAuthInfoHeaders = ApplyFollowerAuthInfoHeaders;
class ApplyFollowerAuthInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appAuthKey: 'appAuthKey',
            fieldScope: 'fieldScope',
            sessionId: 'sessionId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appAuthKey: 'string',
            fieldScope: 'string',
            sessionId: 'string',
            userId: 'string',
        };
    }
}
exports.ApplyFollowerAuthInfoRequest = ApplyFollowerAuthInfoRequest;
class ApplyFollowerAuthInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: ApplyFollowerAuthInfoResponseBodyResult,
        };
    }
}
exports.ApplyFollowerAuthInfoResponseBody = ApplyFollowerAuthInfoResponseBody;
class ApplyFollowerAuthInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ApplyFollowerAuthInfoResponseBody,
        };
    }
}
exports.ApplyFollowerAuthInfoResponse = ApplyFollowerAuthInfoResponse;
class CallbackRegiesterHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CallbackRegiesterHeaders = CallbackRegiesterHeaders;
class CallbackRegiesterRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            apiSecret: 'apiSecret',
            callbackKey: 'callbackKey',
            callbackUrl: 'callbackUrl',
            type: 'type',
        };
    }
    static types() {
        return {
            apiSecret: 'string',
            callbackKey: 'string',
            callbackUrl: 'string',
            type: 'string',
        };
    }
}
exports.CallbackRegiesterRequest = CallbackRegiesterRequest;
class CallbackRegiesterResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: CallbackRegiesterResponseBodyResult,
        };
    }
}
exports.CallbackRegiesterResponseBody = CallbackRegiesterResponseBody;
class CallbackRegiesterResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CallbackRegiesterResponseBody,
        };
    }
}
exports.CallbackRegiesterResponse = CallbackRegiesterResponse;
class CloseTopBoxInteractiveOTOMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CloseTopBoxInteractiveOTOMessageHeaders = CloseTopBoxInteractiveOTOMessageHeaders;
class CloseTopBoxInteractiveOTOMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            detail: 'detail',
        };
    }
    static types() {
        return {
            detail: CloseTopBoxInteractiveOTOMessageRequestDetail,
        };
    }
}
exports.CloseTopBoxInteractiveOTOMessageRequest = CloseTopBoxInteractiveOTOMessageRequest;
class CloseTopBoxInteractiveOTOMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: 'boolean',
        };
    }
}
exports.CloseTopBoxInteractiveOTOMessageResponseBody = CloseTopBoxInteractiveOTOMessageResponseBody;
class CloseTopBoxInteractiveOTOMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CloseTopBoxInteractiveOTOMessageResponseBody,
        };
    }
}
exports.CloseTopBoxInteractiveOTOMessageResponse = CloseTopBoxInteractiveOTOMessageResponse;
class GetFollowerAuthInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetFollowerAuthInfoHeaders = GetFollowerAuthInfoHeaders;
class GetFollowerAuthInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            accountId: 'string',
            userId: 'string',
        };
    }
}
exports.GetFollowerAuthInfoRequest = GetFollowerAuthInfoRequest;
class GetFollowerAuthInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetFollowerAuthInfoResponseBodyResult,
        };
    }
}
exports.GetFollowerAuthInfoResponseBody = GetFollowerAuthInfoResponseBody;
class GetFollowerAuthInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetFollowerAuthInfoResponseBody,
        };
    }
}
exports.GetFollowerAuthInfoResponse = GetFollowerAuthInfoResponse;
class GetFollowerInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetFollowerInfoHeaders = GetFollowerInfoHeaders;
class GetFollowerInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            unionId: 'unionId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            accountId: 'string',
            unionId: 'string',
            userId: 'string',
        };
    }
}
exports.GetFollowerInfoRequest = GetFollowerInfoRequest;
class GetFollowerInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: GetFollowerInfoResponseBodyResult,
        };
    }
}
exports.GetFollowerInfoResponseBody = GetFollowerInfoResponseBody;
class GetFollowerInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetFollowerInfoResponseBody,
        };
    }
}
exports.GetFollowerInfoResponse = GetFollowerInfoResponse;
class GetPictureDownloadUrlHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetPictureDownloadUrlHeaders = GetPictureDownloadUrlHeaders;
class GetPictureDownloadUrlRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            downloadCode: 'downloadCode',
            sessionId: 'sessionId',
        };
    }
    static types() {
        return {
            downloadCode: 'string',
            sessionId: 'string',
        };
    }
}
exports.GetPictureDownloadUrlRequest = GetPictureDownloadUrlRequest;
class GetPictureDownloadUrlResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: GetPictureDownloadUrlResponseBodyResult,
        };
    }
}
exports.GetPictureDownloadUrlResponseBody = GetPictureDownloadUrlResponseBody;
class GetPictureDownloadUrlResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetPictureDownloadUrlResponseBody,
        };
    }
}
exports.GetPictureDownloadUrlResponse = GetPictureDownloadUrlResponse;
class GetUserFollowStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetUserFollowStatusHeaders = GetUserFollowStatusHeaders;
class GetUserFollowStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            unionId: 'unionId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            accountId: 'string',
            unionId: 'string',
            userId: 'string',
        };
    }
}
exports.GetUserFollowStatusRequest = GetUserFollowStatusRequest;
class GetUserFollowStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetUserFollowStatusResponseBodyResult,
        };
    }
}
exports.GetUserFollowStatusResponseBody = GetUserFollowStatusResponseBody;
class GetUserFollowStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetUserFollowStatusResponseBody,
        };
    }
}
exports.GetUserFollowStatusResponse = GetUserFollowStatusResponse;
class ListAccountHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListAccountHeaders = ListAccountHeaders;
class ListAccountResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': ListAccountResponseBodyResult },
        };
    }
}
exports.ListAccountResponseBody = ListAccountResponseBody;
class ListAccountResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListAccountResponseBody,
        };
    }
}
exports.ListAccountResponse = ListAccountResponse;
class ListAccountInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListAccountInfoHeaders = ListAccountInfoHeaders;
class ListAccountInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': ListAccountInfoResponseBodyResult },
        };
    }
}
exports.ListAccountInfoResponseBody = ListAccountInfoResponseBody;
class ListAccountInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListAccountInfoResponseBody,
        };
    }
}
exports.ListAccountInfoResponse = ListAccountInfoResponse;
class ListFollowerHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListFollowerHeaders = ListFollowerHeaders;
class ListFollowerRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            accountId: 'string',
            maxResults: 'number',
            nextToken: 'string',
        };
    }
}
exports.ListFollowerRequest = ListFollowerRequest;
class ListFollowerResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: ListFollowerResponseBodyResult,
        };
    }
}
exports.ListFollowerResponseBody = ListFollowerResponseBody;
class ListFollowerResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListFollowerResponseBody,
        };
    }
}
exports.ListFollowerResponse = ListFollowerResponse;
class QueryUserFollowStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryUserFollowStatusHeaders = QueryUserFollowStatusHeaders;
class QueryUserFollowStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            accountId: 'string',
            unionId: 'string',
        };
    }
}
exports.QueryUserFollowStatusRequest = QueryUserFollowStatusRequest;
class QueryUserFollowStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: QueryUserFollowStatusResponseBodyResult,
        };
    }
}
exports.QueryUserFollowStatusResponseBody = QueryUserFollowStatusResponseBody;
class QueryUserFollowStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryUserFollowStatusResponseBody,
        };
    }
}
exports.QueryUserFollowStatusResponse = QueryUserFollowStatusResponse;
class SendAgentOTOMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendAgentOTOMessageHeaders = SendAgentOTOMessageHeaders;
class SendAgentOTOMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            detail: 'detail',
        };
    }
    static types() {
        return {
            detail: SendAgentOTOMessageRequestDetail,
        };
    }
}
exports.SendAgentOTOMessageRequest = SendAgentOTOMessageRequest;
class SendAgentOTOMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: SendAgentOTOMessageResponseBodyResult,
        };
    }
}
exports.SendAgentOTOMessageResponseBody = SendAgentOTOMessageResponseBody;
class SendAgentOTOMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendAgentOTOMessageResponseBody,
        };
    }
}
exports.SendAgentOTOMessageResponse = SendAgentOTOMessageResponse;
class SendInteractiveOTOMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendInteractiveOTOMessageHeaders = SendInteractiveOTOMessageHeaders;
class SendInteractiveOTOMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            detail: 'detail',
        };
    }
    static types() {
        return {
            detail: SendInteractiveOTOMessageRequestDetail,
        };
    }
}
exports.SendInteractiveOTOMessageRequest = SendInteractiveOTOMessageRequest;
class SendInteractiveOTOMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: SendInteractiveOTOMessageResponseBodyResult,
        };
    }
}
exports.SendInteractiveOTOMessageResponseBody = SendInteractiveOTOMessageResponseBody;
class SendInteractiveOTOMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendInteractiveOTOMessageResponseBody,
        };
    }
}
exports.SendInteractiveOTOMessageResponse = SendInteractiveOTOMessageResponse;
class SendTopBoxInteractiveOTOMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendTopBoxInteractiveOTOMessageHeaders = SendTopBoxInteractiveOTOMessageHeaders;
class SendTopBoxInteractiveOTOMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            detail: 'detail',
        };
    }
    static types() {
        return {
            detail: SendTopBoxInteractiveOTOMessageRequestDetail,
        };
    }
}
exports.SendTopBoxInteractiveOTOMessageRequest = SendTopBoxInteractiveOTOMessageRequest;
class SendTopBoxInteractiveOTOMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: 'boolean',
        };
    }
}
exports.SendTopBoxInteractiveOTOMessageResponseBody = SendTopBoxInteractiveOTOMessageResponseBody;
class SendTopBoxInteractiveOTOMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: SendTopBoxInteractiveOTOMessageResponseBody,
        };
    }
}
exports.SendTopBoxInteractiveOTOMessageResponse = SendTopBoxInteractiveOTOMessageResponse;
class UpdateInteractiveOTOMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateInteractiveOTOMessageHeaders = UpdateInteractiveOTOMessageHeaders;
class UpdateInteractiveOTOMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            detail: 'detail',
        };
    }
    static types() {
        return {
            detail: UpdateInteractiveOTOMessageRequestDetail,
        };
    }
}
exports.UpdateInteractiveOTOMessageRequest = UpdateInteractiveOTOMessageRequest;
class UpdateInteractiveOTOMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
            result: 'result',
        };
    }
    static types() {
        return {
            requestId: 'string',
            result: UpdateInteractiveOTOMessageResponseBodyResult,
        };
    }
}
exports.UpdateInteractiveOTOMessageResponseBody = UpdateInteractiveOTOMessageResponseBody;
class UpdateInteractiveOTOMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateInteractiveOTOMessageResponseBody,
        };
    }
}
exports.UpdateInteractiveOTOMessageResponse = UpdateInteractiveOTOMessageResponse;
class UpdateShortcutsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateShortcutsHeaders = UpdateShortcutsHeaders;
class UpdateShortcutsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            details: 'details',
            sessionId: 'sessionId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            details: { 'type': 'array', 'itemType': UpdateShortcutsRequestDetails },
            sessionId: 'string',
            userId: 'string',
        };
    }
}
exports.UpdateShortcutsRequest = UpdateShortcutsRequest;
class UpdateShortcutsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.UpdateShortcutsResponseBody = UpdateShortcutsResponseBody;
class UpdateShortcutsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateShortcutsResponseBody,
        };
    }
}
exports.UpdateShortcutsResponse = UpdateShortcutsResponse;
class ApplyFollowerAuthInfoResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openApplyId: 'openApplyId',
        };
    }
    static types() {
        return {
            openApplyId: 'string',
        };
    }
}
exports.ApplyFollowerAuthInfoResponseBodyResult = ApplyFollowerAuthInfoResponseBodyResult;
class CallbackRegiesterResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            apiSecret: 'apiSecret',
            callbackUrl: 'callbackUrl',
        };
    }
    static types() {
        return {
            apiSecret: 'string',
            callbackUrl: 'string',
        };
    }
}
exports.CallbackRegiesterResponseBodyResult = CallbackRegiesterResponseBodyResult;
class CloseTopBoxInteractiveOTOMessageRequestDetail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardBizId: 'cardBizId',
            cardTemplateId: 'cardTemplateId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            cardBizId: 'string',
            cardTemplateId: 'string',
            userId: 'string',
        };
    }
}
exports.CloseTopBoxInteractiveOTOMessageRequestDetail = CloseTopBoxInteractiveOTOMessageRequestDetail;
class GetFollowerAuthInfoResponseBodyResultAuthInfoMainCorp extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authorized: 'authorized',
            corpName: 'corpName',
        };
    }
    static types() {
        return {
            authorized: 'boolean',
            corpName: 'string',
        };
    }
}
exports.GetFollowerAuthInfoResponseBodyResultAuthInfoMainCorp = GetFollowerAuthInfoResponseBodyResultAuthInfoMainCorp;
class GetFollowerAuthInfoResponseBodyResultAuthInfoMobile extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authorized: 'authorized',
            mobile: 'mobile',
            stateCode: 'stateCode',
        };
    }
    static types() {
        return {
            authorized: 'boolean',
            mobile: 'string',
            stateCode: 'string',
        };
    }
}
exports.GetFollowerAuthInfoResponseBodyResultAuthInfoMobile = GetFollowerAuthInfoResponseBodyResultAuthInfoMobile;
class GetFollowerAuthInfoResponseBodyResultAuthInfo extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mainCorp: 'mainCorp',
            mobile: 'mobile',
        };
    }
    static types() {
        return {
            mainCorp: GetFollowerAuthInfoResponseBodyResultAuthInfoMainCorp,
            mobile: GetFollowerAuthInfoResponseBodyResultAuthInfoMobile,
        };
    }
}
exports.GetFollowerAuthInfoResponseBodyResultAuthInfo = GetFollowerAuthInfoResponseBodyResultAuthInfo;
class GetFollowerAuthInfoResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authInfo: 'authInfo',
        };
    }
    static types() {
        return {
            authInfo: GetFollowerAuthInfoResponseBodyResultAuthInfo,
        };
    }
}
exports.GetFollowerAuthInfoResponseBodyResult = GetFollowerAuthInfoResponseBodyResult;
class GetFollowerInfoResponseBodyResultUser extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            name: 'name',
            timestamp: 'timestamp',
            userId: 'userId',
        };
    }
    static types() {
        return {
            name: 'string',
            timestamp: 'string',
            userId: 'string',
        };
    }
}
exports.GetFollowerInfoResponseBodyResultUser = GetFollowerInfoResponseBodyResultUser;
class GetFollowerInfoResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            user: 'user',
        };
    }
    static types() {
        return {
            user: GetFollowerInfoResponseBodyResultUser,
        };
    }
}
exports.GetFollowerInfoResponseBodyResult = GetFollowerInfoResponseBodyResult;
class GetPictureDownloadUrlResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            url: 'url',
        };
    }
    static types() {
        return {
            url: 'string',
        };
    }
}
exports.GetPictureDownloadUrlResponseBodyResult = GetPictureDownloadUrlResponseBodyResult;
class GetUserFollowStatusResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            status: 'status',
        };
    }
    static types() {
        return {
            status: 'string',
        };
    }
}
exports.GetUserFollowStatusResponseBodyResult = GetUserFollowStatusResponseBodyResult;
class ListAccountResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            accountName: 'accountName',
        };
    }
    static types() {
        return {
            accountId: 'string',
            accountName: 'string',
        };
    }
}
exports.ListAccountResponseBodyResult = ListAccountResponseBodyResult;
class ListAccountInfoResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            accountId: 'accountId',
            accountName: 'accountName',
        };
    }
    static types() {
        return {
            accountId: 'string',
            accountName: 'string',
        };
    }
}
exports.ListAccountInfoResponseBodyResult = ListAccountInfoResponseBodyResult;
class ListFollowerResponseBodyResultUserList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            name: 'name',
            timestamp: 'timestamp',
            userId: 'userId',
        };
    }
    static types() {
        return {
            name: 'string',
            timestamp: 'number',
            userId: 'string',
        };
    }
}
exports.ListFollowerResponseBodyResultUserList = ListFollowerResponseBodyResultUserList;
class ListFollowerResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            nextToken: 'nextToken',
            userList: 'userList',
        };
    }
    static types() {
        return {
            nextToken: 'string',
            userList: { 'type': 'array', 'itemType': ListFollowerResponseBodyResultUserList },
        };
    }
}
exports.ListFollowerResponseBodyResult = ListFollowerResponseBodyResult;
class QueryUserFollowStatusResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            status: 'status',
        };
    }
    static types() {
        return {
            status: 'string',
        };
    }
}
exports.QueryUserFollowStatusResponseBodyResult = QueryUserFollowStatusResponseBodyResult;
class SendAgentOTOMessageRequestDetailMessageBodyActionCardButtonList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actionUrl: 'actionUrl',
            title: 'title',
        };
    }
    static types() {
        return {
            actionUrl: 'string',
            title: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyActionCardButtonList = SendAgentOTOMessageRequestDetailMessageBodyActionCardButtonList;
class SendAgentOTOMessageRequestDetailMessageBodyActionCard extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            buttonList: 'buttonList',
            buttonOrientation: 'buttonOrientation',
            markdown: 'markdown',
            singleTitle: 'singleTitle',
            singleUrl: 'singleUrl',
            title: 'title',
        };
    }
    static types() {
        return {
            buttonList: { 'type': 'array', 'itemType': SendAgentOTOMessageRequestDetailMessageBodyActionCardButtonList },
            buttonOrientation: 'string',
            markdown: 'string',
            singleTitle: 'string',
            singleUrl: 'string',
            title: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyActionCard = SendAgentOTOMessageRequestDetailMessageBodyActionCard;
class SendAgentOTOMessageRequestDetailMessageBodyImage extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mediaId: 'mediaId',
        };
    }
    static types() {
        return {
            mediaId: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyImage = SendAgentOTOMessageRequestDetailMessageBodyImage;
class SendAgentOTOMessageRequestDetailMessageBodyInteractiveMessage extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            callbackUrl: 'callbackUrl',
            cardBizId: 'cardBizId',
            cardData: 'cardData',
            cardTemplateId: 'cardTemplateId',
        };
    }
    static types() {
        return {
            callbackUrl: 'string',
            cardBizId: 'string',
            cardData: 'string',
            cardTemplateId: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyInteractiveMessage = SendAgentOTOMessageRequestDetailMessageBodyInteractiveMessage;
class SendAgentOTOMessageRequestDetailMessageBodyLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            messageUrl: 'messageUrl',
            picUrl: 'picUrl',
            text: 'text',
            title: 'title',
        };
    }
    static types() {
        return {
            messageUrl: 'string',
            picUrl: 'string',
            text: 'string',
            title: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyLink = SendAgentOTOMessageRequestDetailMessageBodyLink;
class SendAgentOTOMessageRequestDetailMessageBodyMarkdown extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            text: 'text',
            title: 'title',
        };
    }
    static types() {
        return {
            text: 'string',
            title: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyMarkdown = SendAgentOTOMessageRequestDetailMessageBodyMarkdown;
class SendAgentOTOMessageRequestDetailMessageBodyText extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'content',
        };
    }
    static types() {
        return {
            content: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBodyText = SendAgentOTOMessageRequestDetailMessageBodyText;
class SendAgentOTOMessageRequestDetailMessageBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actionCard: 'actionCard',
            image: 'image',
            interactiveMessage: 'interactiveMessage',
            link: 'link',
            markdown: 'markdown',
            text: 'text',
        };
    }
    static types() {
        return {
            actionCard: SendAgentOTOMessageRequestDetailMessageBodyActionCard,
            image: SendAgentOTOMessageRequestDetailMessageBodyImage,
            interactiveMessage: SendAgentOTOMessageRequestDetailMessageBodyInteractiveMessage,
            link: SendAgentOTOMessageRequestDetailMessageBodyLink,
            markdown: SendAgentOTOMessageRequestDetailMessageBodyMarkdown,
            text: SendAgentOTOMessageRequestDetailMessageBodyText,
        };
    }
}
exports.SendAgentOTOMessageRequestDetailMessageBody = SendAgentOTOMessageRequestDetailMessageBody;
class SendAgentOTOMessageRequestDetail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            messageBody: 'messageBody',
            msgType: 'msgType',
            sessionId: 'sessionId',
            userId: 'userId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            messageBody: SendAgentOTOMessageRequestDetailMessageBody,
            msgType: 'string',
            sessionId: 'string',
            userId: 'string',
            uuid: 'string',
        };
    }
}
exports.SendAgentOTOMessageRequestDetail = SendAgentOTOMessageRequestDetail;
class SendAgentOTOMessageResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openPushId: 'openPushId',
        };
    }
    static types() {
        return {
            openPushId: 'string',
        };
    }
}
exports.SendAgentOTOMessageResponseBodyResult = SendAgentOTOMessageResponseBodyResult;
class SendInteractiveOTOMessageRequestDetail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            callbackUrl: 'callbackUrl',
            cardBizId: 'cardBizId',
            cardData: 'cardData',
            cardTemplateId: 'cardTemplateId',
            userId: 'userId',
            userIdPrivateDataMap: 'userIdPrivateDataMap',
        };
    }
    static types() {
        return {
            callbackUrl: 'string',
            cardBizId: 'string',
            cardData: 'string',
            cardTemplateId: 'string',
            userId: 'string',
            userIdPrivateDataMap: 'string',
        };
    }
}
exports.SendInteractiveOTOMessageRequestDetail = SendInteractiveOTOMessageRequestDetail;
class SendInteractiveOTOMessageResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openPushId: 'openPushId',
        };
    }
    static types() {
        return {
            openPushId: 'string',
        };
    }
}
exports.SendInteractiveOTOMessageResponseBodyResult = SendInteractiveOTOMessageResponseBodyResult;
class SendTopBoxInteractiveOTOMessageRequestDetailCardData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardMediaIdParamMap: 'cardMediaIdParamMap',
            cardParamMap: 'cardParamMap',
        };
    }
    static types() {
        return {
            cardMediaIdParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            cardParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.SendTopBoxInteractiveOTOMessageRequestDetailCardData = SendTopBoxInteractiveOTOMessageRequestDetailCardData;
class SendTopBoxInteractiveOTOMessageRequestDetail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            callbackUrl: 'callbackUrl',
            cardBizId: 'cardBizId',
            cardData: 'cardData',
            cardTemplateId: 'cardTemplateId',
            expiredTime: 'expiredTime',
            userId: 'userId',
            userIdPrivateDataMap: 'userIdPrivateDataMap',
        };
    }
    static types() {
        return {
            callbackUrl: 'string',
            cardBizId: 'string',
            cardData: SendTopBoxInteractiveOTOMessageRequestDetailCardData,
            cardTemplateId: 'string',
            expiredTime: 'number',
            userId: 'string',
            userIdPrivateDataMap: { 'type': 'map', 'keyType': 'string', 'valueType': DetailUserIdPrivateDataMapValue },
        };
    }
}
exports.SendTopBoxInteractiveOTOMessageRequestDetail = SendTopBoxInteractiveOTOMessageRequestDetail;
class UpdateInteractiveOTOMessageRequestDetailUpdateOptions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            updateCardDataByKey: 'updateCardDataByKey',
            updatePrivateDataByKey: 'updatePrivateDataByKey',
        };
    }
    static types() {
        return {
            updateCardDataByKey: 'boolean',
            updatePrivateDataByKey: 'boolean',
        };
    }
}
exports.UpdateInteractiveOTOMessageRequestDetailUpdateOptions = UpdateInteractiveOTOMessageRequestDetailUpdateOptions;
class UpdateInteractiveOTOMessageRequestDetail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardBizId: 'cardBizId',
            cardData: 'cardData',
            updateOptions: 'updateOptions',
            userIdPrivateDataMap: 'userIdPrivateDataMap',
        };
    }
    static types() {
        return {
            cardBizId: 'string',
            cardData: 'string',
            updateOptions: UpdateInteractiveOTOMessageRequestDetailUpdateOptions,
            userIdPrivateDataMap: 'string',
        };
    }
}
exports.UpdateInteractiveOTOMessageRequestDetail = UpdateInteractiveOTOMessageRequestDetail;
class UpdateInteractiveOTOMessageResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openPushId: 'openPushId',
        };
    }
    static types() {
        return {
            openPushId: 'string',
        };
    }
}
exports.UpdateInteractiveOTOMessageResponseBodyResult = UpdateInteractiveOTOMessageResponseBodyResult;
class UpdateShortcutsRequestDetails extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actionUrl: 'actionUrl',
            callbackKey: 'callbackKey',
            iconFont: 'iconFont',
            iconMediaId: 'iconMediaId',
            shortcutId: 'shortcutId',
            slideIconMediaId: 'slideIconMediaId',
            title: 'title',
        };
    }
    static types() {
        return {
            actionUrl: 'string',
            callbackKey: 'string',
            iconFont: 'string',
            iconMediaId: 'string',
            shortcutId: 'string',
            slideIconMediaId: 'string',
            title: 'string',
        };
    }
}
exports.UpdateShortcutsRequestDetails = UpdateShortcutsRequestDetails;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async applyFollowerAuthInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appAuthKey)) {
            body["appAuthKey"] = request.appAuthKey;
        }
        if (!tea_util_1.default.isUnset(request.fieldScope)) {
            body["fieldScope"] = request.fieldScope;
        }
        if (!tea_util_1.default.isUnset(request.sessionId)) {
            body["sessionId"] = request.sessionId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ApplyFollowerAuthInfo",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/followers/authInfos/apply`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ApplyFollowerAuthInfoResponse({}));
    }
    async applyFollowerAuthInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ApplyFollowerAuthInfoHeaders({});
        return await this.applyFollowerAuthInfoWithOptions(request, headers, runtime);
    }
    async callbackRegiesterWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.apiSecret)) {
            body["apiSecret"] = request.apiSecret;
        }
        if (!tea_util_1.default.isUnset(request.callbackKey)) {
            body["callbackKey"] = request.callbackKey;
        }
        if (!tea_util_1.default.isUnset(request.callbackUrl)) {
            body["callbackUrl"] = request.callbackUrl;
        }
        if (!tea_util_1.default.isUnset(request.type)) {
            body["type"] = request.type;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CallbackRegiester",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/callbacks/regiester`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CallbackRegiesterResponse({}));
    }
    async callbackRegiester(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CallbackRegiesterHeaders({});
        return await this.callbackRegiesterWithOptions(request, headers, runtime);
    }
    async closeTopBoxInteractiveOTOMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.detail)) {
            body["detail"] = request.detail;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CloseTopBoxInteractiveOTOMessage",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/oToMessages/topBoxes/close`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CloseTopBoxInteractiveOTOMessageResponse({}));
    }
    async closeTopBoxInteractiveOTOMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CloseTopBoxInteractiveOTOMessageHeaders({});
        return await this.closeTopBoxInteractiveOTOMessageWithOptions(request, headers, runtime);
    }
    async getFollowerAuthInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.accountId)) {
            query["accountId"] = request.accountId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetFollowerAuthInfo",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/followers/authInfos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetFollowerAuthInfoResponse({}));
    }
    async getFollowerAuthInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetFollowerAuthInfoHeaders({});
        return await this.getFollowerAuthInfoWithOptions(request, headers, runtime);
    }
    async getFollowerInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.accountId)) {
            query["accountId"] = request.accountId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetFollowerInfo",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/followers/infos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetFollowerInfoResponse({}));
    }
    async getFollowerInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetFollowerInfoHeaders({});
        return await this.getFollowerInfoWithOptions(request, headers, runtime);
    }
    async getPictureDownloadUrlWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.downloadCode)) {
            query["downloadCode"] = request.downloadCode;
        }
        if (!tea_util_1.default.isUnset(request.sessionId)) {
            query["sessionId"] = request.sessionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetPictureDownloadUrl",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/oToMessages/pictures/downloadUrls`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetPictureDownloadUrlResponse({}));
    }
    async getPictureDownloadUrl(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPictureDownloadUrlHeaders({});
        return await this.getPictureDownloadUrlWithOptions(request, headers, runtime);
    }
    async getUserFollowStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.accountId)) {
            query["accountId"] = request.accountId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetUserFollowStatus",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/followers/statuses`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetUserFollowStatusResponse({}));
    }
    async getUserFollowStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserFollowStatusHeaders({});
        return await this.getUserFollowStatusWithOptions(request, headers, runtime);
    }
    async listAccountWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListAccount",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/accounts`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListAccountResponse({}));
    }
    async listAccount() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListAccountHeaders({});
        return await this.listAccountWithOptions(headers, runtime);
    }
    async listAccountInfoWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "ListAccountInfo",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/isv/accounts`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListAccountInfoResponse({}));
    }
    async listAccountInfo() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListAccountInfoHeaders({});
        return await this.listAccountInfoWithOptions(headers, runtime);
    }
    async listFollowerWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.accountId)) {
            query["accountId"] = request.accountId;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListFollower",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/followers`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListFollowerResponse({}));
    }
    async listFollower(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListFollowerHeaders({});
        return await this.listFollowerWithOptions(request, headers, runtime);
    }
    async queryUserFollowStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.accountId)) {
            query["accountId"] = request.accountId;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryUserFollowStatus",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/isv/followers/statuses`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryUserFollowStatusResponse({}));
    }
    async queryUserFollowStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryUserFollowStatusHeaders({});
        return await this.queryUserFollowStatusWithOptions(request, headers, runtime);
    }
    async sendAgentOTOMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.detail)) {
            body["detail"] = request.detail;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendAgentOTOMessage",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/oToMessages/agentMessages`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendAgentOTOMessageResponse({}));
    }
    async sendAgentOTOMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendAgentOTOMessageHeaders({});
        return await this.sendAgentOTOMessageWithOptions(request, headers, runtime);
    }
    async sendInteractiveOTOMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.detail)) {
            body["detail"] = request.detail;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendInteractiveOTOMessage",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/oToMessages/interactiveMessages`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendInteractiveOTOMessageResponse({}));
    }
    async sendInteractiveOTOMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendInteractiveOTOMessageHeaders({});
        return await this.sendInteractiveOTOMessageWithOptions(request, headers, runtime);
    }
    async sendTopBoxInteractiveOTOMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.detail)) {
            body["detail"] = request.detail;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "SendTopBoxInteractiveOTOMessage",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/oToMessages/topBoxes/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new SendTopBoxInteractiveOTOMessageResponse({}));
    }
    async sendTopBoxInteractiveOTOMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendTopBoxInteractiveOTOMessageHeaders({});
        return await this.sendTopBoxInteractiveOTOMessageWithOptions(request, headers, runtime);
    }
    async updateInteractiveOTOMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.detail)) {
            body["detail"] = request.detail;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateInteractiveOTOMessage",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/oToMessages/interactiveMessages`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateInteractiveOTOMessageResponse({}));
    }
    async updateInteractiveOTOMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateInteractiveOTOMessageHeaders({});
        return await this.updateInteractiveOTOMessageWithOptions(request, headers, runtime);
    }
    async updateShortcutsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.details)) {
            body["details"] = request.details;
        }
        if (!tea_util_1.default.isUnset(request.sessionId)) {
            body["sessionId"] = request.sessionId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateShortcuts",
            version: "link_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/link/shortcuts`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateShortcutsResponse({}));
    }
    async updateShortcuts(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateShortcutsHeaders({});
        return await this.updateShortcutsWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
