// @alicloud/dingtalk/dist/cool_ops_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class UpdateIsvOppStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateIsvOppStatusHeaders = UpdateIsvOppStatusHeaders;
class UpdateIsvOppStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            isvOpportunityStatusList: 'isvOpportunityStatusList',
        };
    }
    static types() {
        return {
            isvOpportunityStatusList: { 'type': 'array', 'itemType': UpdateIsvOppStatusRequestIsvOpportunityStatusList },
        };
    }
}
exports.UpdateIsvOppStatusRequest = UpdateIsvOppStatusRequest;
class UpdateIsvOppStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.UpdateIsvOppStatusResponseBody = UpdateIsvOppStatusResponseBody;
class UpdateIsvOppStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateIsvOppStatusResponseBody,
        };
    }
}
exports.UpdateIsvOppStatusResponse = UpdateIsvOppStatusResponse;
class UpdateIsvOppStatusRequestIsvOpportunityStatusList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            isvCorpId: 'isvCorpId',
            microAppId: 'microAppId',
            name: 'name',
            note: 'note',
            operName: 'operName',
            operTime: 'operTime',
            operUserId: 'operUserId',
            oppSourceCorpId: 'oppSourceCorpId',
            opportunityStatus: 'opportunityStatus',
            userId: 'userId',
        };
    }
    static types() {
        return {
            isvCorpId: 'string',
            microAppId: 'string',
            name: 'string',
            note: 'string',
            operName: 'string',
            operTime: 'string',
            operUserId: 'string',
            oppSourceCorpId: 'string',
            opportunityStatus: 'string',
            userId: 'string',
        };
    }
}
exports.UpdateIsvOppStatusRequestIsvOpportunityStatusList = UpdateIsvOppStatusRequestIsvOpportunityStatusList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async updateIsvOppStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.isvOpportunityStatusList)) {
            body["isvOpportunityStatusList"] = request.isvOpportunityStatusList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateIsvOppStatus",
            version: "coolOps_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/coolOps/isvOpportunities/statuses`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateIsvOppStatusResponse({}));
    }
    async updateIsvOppStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateIsvOppStatusHeaders({});
        return await this.updateIsvOppStatusWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
