// @alicloud/dingtalk/dist/check_in_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class GetCheckinRecordByUserHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCheckinRecordByUserHeaders = GetCheckinRecordByUserHeaders;
class GetCheckinRecordByUserRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            operatorUserId: 'operatorUserId',
            startTime: 'startTime',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            endTime: 'number',
            maxResults: 'number',
            nextToken: 'number',
            operatorUserId: 'string',
            startTime: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetCheckinRecordByUserRequest = GetCheckinRecordByUserRequest;
class GetCheckinRecordByUserResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetCheckinRecordByUserResponseBodyResult,
        };
    }
}
exports.GetCheckinRecordByUserResponseBody = GetCheckinRecordByUserResponseBody;
class GetCheckinRecordByUserResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetCheckinRecordByUserResponseBody,
        };
    }
}
exports.GetCheckinRecordByUserResponse = GetCheckinRecordByUserResponse;
class GetCheckinRecordByUserResponseBodyResultPageListCustomDataList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            key: 'key',
            label: 'label',
            value: 'value',
        };
    }
    static types() {
        return {
            key: 'string',
            label: 'string',
            value: 'string',
        };
    }
}
exports.GetCheckinRecordByUserResponseBodyResultPageListCustomDataList = GetCheckinRecordByUserResponseBodyResultPageListCustomDataList;
class GetCheckinRecordByUserResponseBodyResultPageList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            checkinTime: 'checkinTime',
            customDataList: 'customDataList',
            detailPlace: 'detailPlace',
            imageList: 'imageList',
            place: 'place',
            remark: 'remark',
            userId: 'userId',
            visitUser: 'visitUser',
        };
    }
    static types() {
        return {
            checkinTime: 'number',
            customDataList: { 'type': 'array', 'itemType': GetCheckinRecordByUserResponseBodyResultPageListCustomDataList },
            detailPlace: 'string',
            imageList: { 'type': 'array', 'itemType': 'string' },
            place: 'string',
            remark: 'string',
            userId: 'string',
            visitUser: 'string',
        };
    }
}
exports.GetCheckinRecordByUserResponseBodyResultPageList = GetCheckinRecordByUserResponseBodyResultPageList;
class GetCheckinRecordByUserResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            nextToken: 'nextToken',
            pageList: 'pageList',
        };
    }
    static types() {
        return {
            nextToken: 'number',
            pageList: { 'type': 'array', 'itemType': GetCheckinRecordByUserResponseBodyResultPageList },
        };
    }
}
exports.GetCheckinRecordByUserResponseBodyResult = GetCheckinRecordByUserResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async getCheckinRecordByUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            body["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            body["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.operatorUserId)) {
            body["operatorUserId"] = request.operatorUserId;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetCheckinRecordByUser",
            version: "checkIn_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/checkIn/records/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetCheckinRecordByUserResponse({}));
    }
    async getCheckinRecordByUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCheckinRecordByUserHeaders({});
        return await this.getCheckinRecordByUserWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
