// @alicloud/dingtalk/dist/wiki_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class WikiWordsDetailHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.WikiWordsDetailHeaders = WikiWordsDetailHeaders;
class WikiWordsDetailRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            wordName: 'wordName',
        };
    }
    static types() {
        return {
            wordName: 'string',
        };
    }
}
exports.WikiWordsDetailRequest = WikiWordsDetailRequest;
class WikiWordsDetailResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            data: 'data',
            errMsg: 'errMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            data: { 'type': 'array', 'itemType': WikiWordsDetailResponseBodyData },
            errMsg: 'string',
            success: 'boolean',
        };
    }
}
exports.WikiWordsDetailResponseBody = WikiWordsDetailResponseBody;
class WikiWordsDetailResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: WikiWordsDetailResponseBody,
        };
    }
}
exports.WikiWordsDetailResponse = WikiWordsDetailResponse;
class WikiWordsParseHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.WikiWordsParseHeaders = WikiWordsParseHeaders;
class WikiWordsParseRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            content: 'content',
        };
    }
    static types() {
        return {
            content: 'string',
        };
    }
}
exports.WikiWordsParseRequest = WikiWordsParseRequest;
class WikiWordsParseResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            data: 'data',
            errMsg: 'errMsg',
            success: 'success',
        };
    }
    static types() {
        return {
            data: { 'type': 'array', 'itemType': WikiWordsParseResponseBodyData },
            errMsg: 'string',
            success: 'boolean',
        };
    }
}
exports.WikiWordsParseResponseBody = WikiWordsParseResponseBody;
class WikiWordsParseResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: WikiWordsParseResponseBody,
        };
    }
}
exports.WikiWordsParseResponse = WikiWordsParseResponse;
class WikiWordsDetailResponseBodyDataAppLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appId: 'appId',
            appName: 'appName',
            iconLink: 'iconLink',
            pcLink: 'pcLink',
            phoneLink: 'phoneLink',
        };
    }
    static types() {
        return {
            appId: 'number',
            appName: 'string',
            iconLink: 'string',
            pcLink: 'string',
            phoneLink: 'string',
        };
    }
}
exports.WikiWordsDetailResponseBodyDataAppLink = WikiWordsDetailResponseBodyDataAppLink;
class WikiWordsDetailResponseBodyDataRelatedDoc extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.WikiWordsDetailResponseBodyDataRelatedDoc = WikiWordsDetailResponseBodyDataRelatedDoc;
class WikiWordsDetailResponseBodyDataRelatedLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.WikiWordsDetailResponseBodyDataRelatedLink = WikiWordsDetailResponseBodyDataRelatedLink;
class WikiWordsDetailResponseBodyData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appLink: 'appLink',
            approveName: 'approveName',
            contacts: 'contacts',
            creatorName: 'creatorName',
            gmtCreate: 'gmtCreate',
            gmtModify: 'gmtModify',
            highLightWordAlias: 'highLightWordAlias',
            imHighLight: 'imHighLight',
            orgName: 'orgName',
            relatedDoc: 'relatedDoc',
            relatedLink: 'relatedLink',
            simHighLight: 'simHighLight',
            simpleWordParaphrase: 'simpleWordParaphrase',
            tagsList: 'tagsList',
            updaterName: 'updaterName',
            uuid: 'uuid',
            wordAlias: 'wordAlias',
            wordFullName: 'wordFullName',
            wordName: 'wordName',
            wordParaphrase: 'wordParaphrase',
        };
    }
    static types() {
        return {
            appLink: { 'type': 'array', 'itemType': WikiWordsDetailResponseBodyDataAppLink },
            approveName: 'string',
            contacts: { 'type': 'array', 'itemType': 'string' },
            creatorName: 'string',
            gmtCreate: 'number',
            gmtModify: 'number',
            highLightWordAlias: { 'type': 'array', 'itemType': 'string' },
            imHighLight: 'boolean',
            orgName: 'string',
            relatedDoc: { 'type': 'array', 'itemType': WikiWordsDetailResponseBodyDataRelatedDoc },
            relatedLink: { 'type': 'array', 'itemType': WikiWordsDetailResponseBodyDataRelatedLink },
            simHighLight: 'boolean',
            simpleWordParaphrase: 'string',
            tagsList: { 'type': 'array', 'itemType': 'string' },
            updaterName: 'string',
            uuid: 'number',
            wordAlias: { 'type': 'array', 'itemType': 'string' },
            wordFullName: 'string',
            wordName: 'string',
            wordParaphrase: 'string',
        };
    }
}
exports.WikiWordsDetailResponseBodyData = WikiWordsDetailResponseBodyData;
class WikiWordsParseResponseBodyData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endIndex: 'endIndex',
            startIndex: 'startIndex',
            wordName: 'wordName',
        };
    }
    static types() {
        return {
            endIndex: 'number',
            startIndex: 'number',
            wordName: 'string',
        };
    }
}
exports.WikiWordsParseResponseBodyData = WikiWordsParseResponseBodyData;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async wikiWordsDetailWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.wordName)) {
            query["wordName"] = request.wordName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "WikiWordsDetail",
            version: "wiki_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/wiki/words/details`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new WikiWordsDetailResponse({}));
    }
    async wikiWordsDetail(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new WikiWordsDetailHeaders({});
        return await this.wikiWordsDetailWithOptions(request, headers, runtime);
    }
    async wikiWordsParseWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.content)) {
            body["content"] = request.content;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "WikiWordsParse",
            version: "wiki_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/wiki/words/parse`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new WikiWordsParseResponse({}));
    }
    async wikiWordsParse(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new WikiWordsParseHeaders({});
        return await this.wikiWordsParseWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
