// @alicloud/dingtalk/dist/flashmeeting_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateFlashMeetingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateFlashMeetingHeaders = CreateFlashMeetingHeaders;
class CreateFlashMeetingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            creator: 'creator',
            eventId: 'eventId',
            title: 'title',
        };
    }
    static types() {
        return {
            creator: 'string',
            eventId: 'string',
            title: 'string',
        };
    }
}
exports.CreateFlashMeetingRequest = CreateFlashMeetingRequest;
class CreateFlashMeetingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            flashMeetingKey: 'flashMeetingKey',
            startTime: 'startTime',
            title: 'title',
            url: 'url',
        };
    }
    static types() {
        return {
            endTime: 'number',
            flashMeetingKey: 'string',
            startTime: 'number',
            title: 'string',
            url: 'string',
        };
    }
}
exports.CreateFlashMeetingResponseBody = CreateFlashMeetingResponseBody;
class CreateFlashMeetingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateFlashMeetingResponseBody,
        };
    }
}
exports.CreateFlashMeetingResponse = CreateFlashMeetingResponse;
class GetShanhuiByCalendarHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetShanhuiByCalendarHeaders = GetShanhuiByCalendarHeaders;
class GetShanhuiByCalendarRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            eventId: 'eventId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            eventId: 'string',
            userId: 'string',
        };
    }
}
exports.GetShanhuiByCalendarRequest = GetShanhuiByCalendarRequest;
class GetShanhuiByCalendarResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: GetShanhuiByCalendarResponseBodyResult,
            success: 'boolean',
        };
    }
}
exports.GetShanhuiByCalendarResponseBody = GetShanhuiByCalendarResponseBody;
class GetShanhuiByCalendarResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetShanhuiByCalendarResponseBody,
        };
    }
}
exports.GetShanhuiByCalendarResponse = GetShanhuiByCalendarResponse;
class GetShanhuiByShanhuiKeyHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetShanhuiByShanhuiKeyHeaders = GetShanhuiByShanhuiKeyHeaders;
class GetShanhuiByShanhuiKeyResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: GetShanhuiByShanhuiKeyResponseBodyResult,
            success: 'boolean',
        };
    }
}
exports.GetShanhuiByShanhuiKeyResponseBody = GetShanhuiByShanhuiKeyResponseBody;
class GetShanhuiByShanhuiKeyResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetShanhuiByShanhuiKeyResponseBody,
        };
    }
}
exports.GetShanhuiByShanhuiKeyResponse = GetShanhuiByShanhuiKeyResponse;
class GetTaskFromShanhuiDocHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetTaskFromShanhuiDocHeaders = GetTaskFromShanhuiDocHeaders;
class GetTaskFromShanhuiDocRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            docKey: 'docKey',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            docKey: 'string',
            maxResults: 'number',
            nextToken: 'number',
            unionId: 'string',
        };
    }
}
exports.GetTaskFromShanhuiDocRequest = GetTaskFromShanhuiDocRequest;
class GetTaskFromShanhuiDocResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: GetTaskFromShanhuiDocResponseBodyResult,
            success: 'boolean',
        };
    }
}
exports.GetTaskFromShanhuiDocResponseBody = GetTaskFromShanhuiDocResponseBody;
class GetTaskFromShanhuiDocResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetTaskFromShanhuiDocResponseBody,
        };
    }
}
exports.GetTaskFromShanhuiDocResponse = GetTaskFromShanhuiDocResponse;
class GetShanhuiByCalendarResponseBodyResultTopics extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            docKey: 'docKey',
            title: 'title',
        };
    }
    static types() {
        return {
            docKey: 'string',
            title: 'string',
        };
    }
}
exports.GetShanhuiByCalendarResponseBodyResultTopics = GetShanhuiByCalendarResponseBodyResultTopics;
class GetShanhuiByCalendarResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            flashmeetingKey: 'flashmeetingKey',
            hasSummary: 'hasSummary',
            startTime: 'startTime',
            summaryDocKey: 'summaryDocKey',
            title: 'title',
            topics: 'topics',
        };
    }
    static types() {
        return {
            endTime: 'number',
            flashmeetingKey: 'string',
            hasSummary: 'boolean',
            startTime: 'number',
            summaryDocKey: 'string',
            title: 'string',
            topics: { 'type': 'array', 'itemType': GetShanhuiByCalendarResponseBodyResultTopics },
        };
    }
}
exports.GetShanhuiByCalendarResponseBodyResult = GetShanhuiByCalendarResponseBodyResult;
class GetShanhuiByShanhuiKeyResponseBodyResultTopics extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            docKey: 'docKey',
            title: 'title',
        };
    }
    static types() {
        return {
            docKey: 'string',
            title: 'string',
        };
    }
}
exports.GetShanhuiByShanhuiKeyResponseBodyResultTopics = GetShanhuiByShanhuiKeyResponseBodyResultTopics;
class GetShanhuiByShanhuiKeyResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            eventId: 'eventId',
            flashmeetingKey: 'flashmeetingKey',
            hasSummary: 'hasSummary',
            startTime: 'startTime',
            summaryDocKey: 'summaryDocKey',
            title: 'title',
            topics: 'topics',
        };
    }
    static types() {
        return {
            endTime: 'number',
            eventId: 'string',
            flashmeetingKey: 'string',
            hasSummary: 'boolean',
            startTime: 'number',
            summaryDocKey: 'string',
            title: 'string',
            topics: { 'type': 'array', 'itemType': GetShanhuiByShanhuiKeyResponseBodyResultTopics },
        };
    }
}
exports.GetShanhuiByShanhuiKeyResponseBodyResult = GetShanhuiByShanhuiKeyResponseBodyResult;
class GetTaskFromShanhuiDocResponseBodyResultItems extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            createTime: 'createTime',
            deadline: 'deadline',
            deleted: 'deleted',
            priority: 'priority',
            taskKey: 'taskKey',
            taskStatus: 'taskStatus',
            taskType: 'taskType',
            title: 'title',
            updateTime: 'updateTime',
        };
    }
    static types() {
        return {
            createTime: 'number',
            deadline: 'number',
            deleted: 'boolean',
            priority: 'number',
            taskKey: 'string',
            taskStatus: 'string',
            taskType: 'string',
            title: 'string',
            updateTime: 'number',
        };
    }
}
exports.GetTaskFromShanhuiDocResponseBodyResultItems = GetTaskFromShanhuiDocResponseBodyResultItems;
class GetTaskFromShanhuiDocResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            items: 'items',
            nextToken: 'nextToken',
            total: 'total',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            items: { 'type': 'array', 'itemType': GetTaskFromShanhuiDocResponseBodyResultItems },
            nextToken: 'string',
            total: 'number',
        };
    }
}
exports.GetTaskFromShanhuiDocResponseBodyResult = GetTaskFromShanhuiDocResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async createFlashMeetingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.creator)) {
            body["creator"] = request.creator;
        }
        if (!tea_util_1.default.isUnset(request.eventId)) {
            body["eventId"] = request.eventId;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateFlashMeeting",
            version: "flashmeeting_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmeeting/meetings`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateFlashMeetingResponse({}));
    }
    async createFlashMeeting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateFlashMeetingHeaders({});
        return await this.createFlashMeetingWithOptions(request, headers, runtime);
    }
    async getShanhuiByCalendarWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.eventId)) {
            query["eventId"] = request.eventId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetShanhuiByCalendar",
            version: "flashmeeting_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmeeting/calendars/meeting`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetShanhuiByCalendarResponse({}));
    }
    async getShanhuiByCalendar(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetShanhuiByCalendarHeaders({});
        return await this.getShanhuiByCalendarWithOptions(request, headers, runtime);
    }
    async getShanhuiByShanhuiKeyWithOptions(flashmeetingKey, headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        let params = new $OpenApi.Params({
            action: "GetShanhuiByShanhuiKey",
            version: "flashmeeting_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmeeting/meetings/keys/${flashmeetingKey}/infos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetShanhuiByShanhuiKeyResponse({}));
    }
    async getShanhuiByShanhuiKey(flashmeetingKey) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetShanhuiByShanhuiKeyHeaders({});
        return await this.getShanhuiByShanhuiKeyWithOptions(flashmeetingKey, headers, runtime);
    }
    async getTaskFromShanhuiDocWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.docKey)) {
            query["docKey"] = request.docKey;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetTaskFromShanhuiDoc",
            version: "flashmeeting_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/flashmeeting/meetings/tasks`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetTaskFromShanhuiDocResponse({}));
    }
    async getTaskFromShanhuiDoc(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetTaskFromShanhuiDocHeaders({});
        return await this.getTaskFromShanhuiDocWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
