// @alicloud/dingtalk/dist/activity_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateActivityHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateActivityHeaders = CreateActivityHeaders;
class CreateActivityRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            detail: 'detail',
        };
    }
    static types() {
        return {
            detail: CreateActivityRequestDetail,
        };
    }
}
exports.CreateActivityRequest = CreateActivityRequest;
class CreateActivityResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            activityId: 'activityId',
        };
    }
    static types() {
        return {
            activityId: 'string',
        };
    }
}
exports.CreateActivityResponseBody = CreateActivityResponseBody;
class CreateActivityResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: CreateActivityResponseBody,
        };
    }
}
exports.CreateActivityResponse = CreateActivityResponse;
class ListActivityHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListActivityHeaders = ListActivityHeaders;
class ListActivityRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'string',
        };
    }
}
exports.ListActivityRequest = ListActivityRequest;
class ListActivityResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            list: 'list',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            list: { 'type': 'array', 'itemType': ListActivityResponseBodyList },
            maxResults: 'string',
            nextToken: 'string',
        };
    }
}
exports.ListActivityResponseBody = ListActivityResponseBody;
class ListActivityResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ListActivityResponseBody,
        };
    }
}
exports.ListActivityResponse = ListActivityResponse;
class CreateActivityRequestDetailAddress extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            district: 'district',
            lat: 'lat',
            lng: 'lng',
            name: 'name',
        };
    }
    static types() {
        return {
            district: 'string',
            lat: 'string',
            lng: 'string',
            name: 'string',
        };
    }
}
exports.CreateActivityRequestDetailAddress = CreateActivityRequestDetailAddress;
class CreateActivityRequestDetail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            address: 'address',
            bannerMediaId: 'bannerMediaId',
            endTime: 'endTime',
            foreignId: 'foreignId',
            industry: 'industry',
            roleName: 'roleName',
            source: 'source',
            startTime: 'startTime',
            title: 'title',
            type: 'type',
            url: 'url',
        };
    }
    static types() {
        return {
            address: CreateActivityRequestDetailAddress,
            bannerMediaId: 'string',
            endTime: 'number',
            foreignId: 'string',
            industry: 'string',
            roleName: 'string',
            source: 'string',
            startTime: 'number',
            title: 'string',
            type: 'number',
            url: 'string',
        };
    }
}
exports.CreateActivityRequestDetail = CreateActivityRequestDetail;
class ListActivityResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            activityId: 'activityId',
            bannerMediaId: 'bannerMediaId',
            endTime: 'endTime',
            foreignId: 'foreignId',
            startTime: 'startTime',
            status: 'status',
            title: 'title',
            type: 'type',
            url: 'url',
        };
    }
    static types() {
        return {
            activityId: 'string',
            bannerMediaId: 'string',
            endTime: 'number',
            foreignId: 'string',
            startTime: 'number',
            status: 'string',
            title: 'string',
            type: 'string',
            url: 'string',
        };
    }
}
exports.ListActivityResponseBodyList = ListActivityResponseBodyList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async createActivityWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.detail)) {
            body["detail"] = request.detail;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "CreateActivity",
            version: "activity_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/activity/meta`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new CreateActivityResponse({}));
    }
    async createActivity(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateActivityHeaders({});
        return await this.createActivityWithOptions(request, headers, runtime);
    }
    async listActivityWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "ListActivity",
            version: "activity_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/activity/metaLists`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ListActivityResponse({}));
    }
    async listActivity(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListActivityHeaders({});
        return await this.listActivityWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
