// @alicloud/dingtalk/dist/dingmi_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddRobotInstanceToGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AddRobotInstanceToGroupHeaders = AddRobotInstanceToGroupHeaders;
class AddRobotInstanceToGroupRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            chatbotId: 'chatbotId',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            chatbotId: 'string',
            openConversationId: 'string',
        };
    }
}
exports.AddRobotInstanceToGroupRequest = AddRobotInstanceToGroupRequest;
class AddRobotInstanceToGroupResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.AddRobotInstanceToGroupResponseBody = AddRobotInstanceToGroupResponseBody;
class AddRobotInstanceToGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AddRobotInstanceToGroupResponseBody,
        };
    }
}
exports.AddRobotInstanceToGroupResponse = AddRobotInstanceToGroupResponse;
class AskRobotHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AskRobotHeaders = AskRobotHeaders;
class AskRobotRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            dingUserId: 'dingUserId',
            question: 'question',
            robotAppKey: 'robotAppKey',
            sessionUuid: 'sessionUuid',
        };
    }
    static types() {
        return {
            dingUserId: 'string',
            question: 'string',
            robotAppKey: 'string',
            sessionUuid: 'string',
        };
    }
}
exports.AskRobotRequest = AskRobotRequest;
class AskRobotResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.AskRobotResponseBody = AskRobotResponseBody;
class AskRobotResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AskRobotResponseBody,
        };
    }
}
exports.AskRobotResponse = AskRobotResponse;
class GetDingMeBaseDataHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetDingMeBaseDataHeaders = GetDingMeBaseDataHeaders;
class GetDingMeBaseDataRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appKey: 'appKey',
            byDay: 'byDay',
            endDay: 'endDay',
            startDay: 'startDay',
        };
    }
    static types() {
        return {
            appKey: 'string',
            byDay: 'boolean',
            endDay: 'string',
            startDay: 'string',
        };
    }
}
exports.GetDingMeBaseDataRequest = GetDingMeBaseDataRequest;
class GetDingMeBaseDataResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fromCache: 'fromCache',
            rawset: 'rawset',
            runtime: 'runtime',
            tips: 'tips',
        };
    }
    static types() {
        return {
            fromCache: 'boolean',
            rawset: { 'type': 'array', 'itemType': { 'type': 'map', 'keyType': 'string', 'valueType': 'string' } },
            runtime: 'number',
            tips: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.GetDingMeBaseDataResponseBody = GetDingMeBaseDataResponseBody;
class GetDingMeBaseDataResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetDingMeBaseDataResponseBody,
        };
    }
}
exports.GetDingMeBaseDataResponse = GetDingMeBaseDataResponse;
class GetIntelligentRobotInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetIntelligentRobotInfoHeaders = GetIntelligentRobotInfoHeaders;
class GetIntelligentRobotInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            robotAppKey: 'robotAppKey',
        };
    }
    static types() {
        return {
            robotAppKey: 'string',
        };
    }
}
exports.GetIntelligentRobotInfoRequest = GetIntelligentRobotInfoRequest;
class GetIntelligentRobotInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.GetIntelligentRobotInfoResponseBody = GetIntelligentRobotInfoResponseBody;
class GetIntelligentRobotInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetIntelligentRobotInfoResponseBody,
        };
    }
}
exports.GetIntelligentRobotInfoResponse = GetIntelligentRobotInfoResponse;
class GetOfficialAccountRobotInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetOfficialAccountRobotInfoHeaders = GetOfficialAccountRobotInfoHeaders;
class GetOfficialAccountRobotInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            type: 'type',
        };
    }
    static types() {
        return {
            type: 'string',
        };
    }
}
exports.GetOfficialAccountRobotInfoRequest = GetOfficialAccountRobotInfoRequest;
class GetOfficialAccountRobotInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appId: 'appId',
            brief: 'brief',
            description: 'description',
            icon: 'icon',
            name: 'name',
            previewMediaUrl: 'previewMediaUrl',
        };
    }
    static types() {
        return {
            appId: 'number',
            brief: 'string',
            description: 'string',
            icon: 'string',
            name: 'string',
            previewMediaUrl: 'string',
        };
    }
}
exports.GetOfficialAccountRobotInfoResponseBody = GetOfficialAccountRobotInfoResponseBody;
class GetOfficialAccountRobotInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetOfficialAccountRobotInfoResponseBody,
        };
    }
}
exports.GetOfficialAccountRobotInfoResponse = GetOfficialAccountRobotInfoResponse;
class GetWebChannelUserTokenHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetWebChannelUserTokenHeaders = GetWebChannelUserTokenHeaders;
class GetWebChannelUserTokenRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            foreignId: 'foreignId',
            nick: 'nick',
            source: 'source',
        };
    }
    static types() {
        return {
            foreignId: 'string',
            nick: 'string',
            source: 'number',
        };
    }
}
exports.GetWebChannelUserTokenRequest = GetWebChannelUserTokenRequest;
class GetWebChannelUserTokenResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.GetWebChannelUserTokenResponseBody = GetWebChannelUserTokenResponseBody;
class GetWebChannelUserTokenResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: GetWebChannelUserTokenResponseBody,
        };
    }
}
exports.GetWebChannelUserTokenResponse = GetWebChannelUserTokenResponse;
class PushCustomerGroupMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PushCustomerGroupMessageHeaders = PushCustomerGroupMessageHeaders;
class PushCustomerGroupMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            conversationId: 'conversationId',
            msgKey: 'msgKey',
            msgParam: 'msgParam',
        };
    }
    static types() {
        return {
            conversationId: 'string',
            msgKey: 'string',
            msgParam: 'string',
        };
    }
}
exports.PushCustomerGroupMessageRequest = PushCustomerGroupMessageRequest;
class PushCustomerGroupMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.PushCustomerGroupMessageResponseBody = PushCustomerGroupMessageResponseBody;
class PushCustomerGroupMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushCustomerGroupMessageResponseBody,
        };
    }
}
exports.PushCustomerGroupMessageResponse = PushCustomerGroupMessageResponse;
class PushIntelligentRobotGroupMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PushIntelligentRobotGroupMessageHeaders = PushIntelligentRobotGroupMessageHeaders;
class PushIntelligentRobotGroupMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            chatbotId: 'chatbotId',
            msgKey: 'msgKey',
            msgParam: 'msgParam',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            chatbotId: 'string',
            msgKey: 'string',
            msgParam: 'string',
            openConversationId: 'string',
        };
    }
}
exports.PushIntelligentRobotGroupMessageRequest = PushIntelligentRobotGroupMessageRequest;
class PushIntelligentRobotGroupMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.PushIntelligentRobotGroupMessageResponseBody = PushIntelligentRobotGroupMessageResponseBody;
class PushIntelligentRobotGroupMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushIntelligentRobotGroupMessageResponseBody,
        };
    }
}
exports.PushIntelligentRobotGroupMessageResponse = PushIntelligentRobotGroupMessageResponse;
class PushIntelligentRobotMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PushIntelligentRobotMessageHeaders = PushIntelligentRobotMessageHeaders;
class PushIntelligentRobotMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            chatbotId: 'chatbotId',
            msgKey: 'msgKey',
            msgParam: 'msgParam',
            userId: 'userId',
        };
    }
    static types() {
        return {
            chatbotId: 'string',
            msgKey: 'string',
            msgParam: 'string',
            userId: 'string',
        };
    }
}
exports.PushIntelligentRobotMessageRequest = PushIntelligentRobotMessageRequest;
class PushIntelligentRobotMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.PushIntelligentRobotMessageResponseBody = PushIntelligentRobotMessageResponseBody;
class PushIntelligentRobotMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushIntelligentRobotMessageResponseBody,
        };
    }
}
exports.PushIntelligentRobotMessageResponse = PushIntelligentRobotMessageResponse;
class PushOfficialAccountMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PushOfficialAccountMessageHeaders = PushOfficialAccountMessageHeaders;
class PushOfficialAccountMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            msgKey: 'msgKey',
            msgParam: 'msgParam',
            userId: 'userId',
        };
    }
    static types() {
        return {
            msgKey: 'string',
            msgParam: 'string',
            userId: 'string',
        };
    }
}
exports.PushOfficialAccountMessageRequest = PushOfficialAccountMessageRequest;
class PushOfficialAccountMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.PushOfficialAccountMessageResponseBody = PushOfficialAccountMessageResponseBody;
class PushOfficialAccountMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushOfficialAccountMessageResponseBody,
        };
    }
}
exports.PushOfficialAccountMessageResponse = PushOfficialAccountMessageResponse;
class PushRobotMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PushRobotMessageHeaders = PushRobotMessageHeaders;
class PushRobotMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            chatbotId: 'chatbotId',
            msgKey: 'msgKey',
            msgParam: 'msgParam',
            userId: 'userId',
        };
    }
    static types() {
        return {
            chatbotId: 'string',
            msgKey: 'string',
            msgParam: 'string',
            userId: 'string',
        };
    }
}
exports.PushRobotMessageRequest = PushRobotMessageRequest;
class PushRobotMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.PushRobotMessageResponseBody = PushRobotMessageResponseBody;
class PushRobotMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushRobotMessageResponseBody,
        };
    }
}
exports.PushRobotMessageResponse = PushRobotMessageResponse;
class ReplyRobotHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ReplyRobotHeaders = ReplyRobotHeaders;
class ReplyRobotRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            proxyMessageStr: 'proxyMessageStr',
        };
    }
    static types() {
        return {
            proxyMessageStr: 'string',
        };
    }
}
exports.ReplyRobotRequest = ReplyRobotRequest;
class ReplyRobotResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.ReplyRobotResponseBody = ReplyRobotResponseBody;
class ReplyRobotResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: ReplyRobotResponseBody,
        };
    }
}
exports.ReplyRobotResponse = ReplyRobotResponse;
class UpdateOfficialAccountRobotInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateOfficialAccountRobotInfoHeaders = UpdateOfficialAccountRobotInfoHeaders;
class UpdateOfficialAccountRobotInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatar: 'avatar',
            brief: 'brief',
            description: 'description',
            name: 'name',
            previewMediaUrl: 'previewMediaUrl',
            type: 'type',
        };
    }
    static types() {
        return {
            avatar: 'string',
            brief: 'string',
            description: 'string',
            name: 'string',
            previewMediaUrl: 'string',
            type: 'string',
        };
    }
}
exports.UpdateOfficialAccountRobotInfoRequest = UpdateOfficialAccountRobotInfoRequest;
class UpdateOfficialAccountRobotInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.UpdateOfficialAccountRobotInfoResponseBody = UpdateOfficialAccountRobotInfoResponseBody;
class UpdateOfficialAccountRobotInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpdateOfficialAccountRobotInfoResponseBody,
        };
    }
}
exports.UpdateOfficialAccountRobotInfoResponse = UpdateOfficialAccountRobotInfoResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async addRobotInstanceToGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.chatbotId)) {
            body["chatbotId"] = request.chatbotId;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AddRobotInstanceToGroup",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/intelligentRobots/groups`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AddRobotInstanceToGroupResponse({}));
    }
    async addRobotInstanceToGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddRobotInstanceToGroupHeaders({});
        return await this.addRobotInstanceToGroupWithOptions(request, headers, runtime);
    }
    async askRobotWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.dingUserId)) {
            body["dingUserId"] = request.dingUserId;
        }
        if (!tea_util_1.default.isUnset(request.question)) {
            body["question"] = request.question;
        }
        if (!tea_util_1.default.isUnset(request.robotAppKey)) {
            body["robotAppKey"] = request.robotAppKey;
        }
        if (!tea_util_1.default.isUnset(request.sessionUuid)) {
            body["sessionUuid"] = request.sessionUuid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "AskRobot",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/robots/ask`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AskRobotResponse({}));
    }
    async askRobot(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AskRobotHeaders({});
        return await this.askRobotWithOptions(request, headers, runtime);
    }
    async getDingMeBaseDataWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.appKey)) {
            query["appKey"] = request.appKey;
        }
        if (!tea_util_1.default.isUnset(request.byDay)) {
            query["byDay"] = request.byDay;
        }
        if (!tea_util_1.default.isUnset(request.endDay)) {
            query["endDay"] = request.endDay;
        }
        if (!tea_util_1.default.isUnset(request.startDay)) {
            query["startDay"] = request.startDay;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetDingMeBaseData",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/robots/data`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetDingMeBaseDataResponse({}));
    }
    async getDingMeBaseData(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetDingMeBaseDataHeaders({});
        return await this.getDingMeBaseDataWithOptions(request, headers, runtime);
    }
    async getIntelligentRobotInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.robotAppKey)) {
            query["robotAppKey"] = request.robotAppKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetIntelligentRobotInfo",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/intelligentRobots/infos`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetIntelligentRobotInfoResponse({}));
    }
    async getIntelligentRobotInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetIntelligentRobotInfoHeaders({});
        return await this.getIntelligentRobotInfoWithOptions(request, headers, runtime);
    }
    async getOfficialAccountRobotInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.type)) {
            query["type"] = request.type;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "GetOfficialAccountRobotInfo",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/officialAccounts/robots`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetOfficialAccountRobotInfoResponse({}));
    }
    async getOfficialAccountRobotInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetOfficialAccountRobotInfoHeaders({});
        return await this.getOfficialAccountRobotInfoWithOptions(request, headers, runtime);
    }
    async getWebChannelUserTokenWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.foreignId)) {
            body["foreignId"] = request.foreignId;
        }
        if (!tea_util_1.default.isUnset(request.nick)) {
            body["nick"] = request.nick;
        }
        if (!tea_util_1.default.isUnset(request.source)) {
            body["source"] = request.source;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "GetWebChannelUserToken",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/webChannels/userTokens`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new GetWebChannelUserTokenResponse({}));
    }
    async getWebChannelUserToken(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetWebChannelUserTokenHeaders({});
        return await this.getWebChannelUserTokenWithOptions(request, headers, runtime);
    }
    async pushCustomerGroupMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.conversationId)) {
            body["conversationId"] = request.conversationId;
        }
        if (!tea_util_1.default.isUnset(request.msgKey)) {
            body["msgKey"] = request.msgKey;
        }
        if (!tea_util_1.default.isUnset(request.msgParam)) {
            body["msgParam"] = request.msgParam;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushCustomerGroupMessage",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/officialAccounts/robots/groupMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushCustomerGroupMessageResponse({}));
    }
    async pushCustomerGroupMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushCustomerGroupMessageHeaders({});
        return await this.pushCustomerGroupMessageWithOptions(request, headers, runtime);
    }
    async pushIntelligentRobotGroupMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.chatbotId)) {
            body["chatbotId"] = request.chatbotId;
        }
        if (!tea_util_1.default.isUnset(request.msgKey)) {
            body["msgKey"] = request.msgKey;
        }
        if (!tea_util_1.default.isUnset(request.msgParam)) {
            body["msgParam"] = request.msgParam;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushIntelligentRobotGroupMessage",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/intelligentRobots/groupMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushIntelligentRobotGroupMessageResponse({}));
    }
    async pushIntelligentRobotGroupMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushIntelligentRobotGroupMessageHeaders({});
        return await this.pushIntelligentRobotGroupMessageWithOptions(request, headers, runtime);
    }
    async pushIntelligentRobotMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.chatbotId)) {
            body["chatbotId"] = request.chatbotId;
        }
        if (!tea_util_1.default.isUnset(request.msgKey)) {
            body["msgKey"] = request.msgKey;
        }
        if (!tea_util_1.default.isUnset(request.msgParam)) {
            body["msgParam"] = request.msgParam;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushIntelligentRobotMessage",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/intelligentRobots/oToMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushIntelligentRobotMessageResponse({}));
    }
    async pushIntelligentRobotMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushIntelligentRobotMessageHeaders({});
        return await this.pushIntelligentRobotMessageWithOptions(request, headers, runtime);
    }
    async pushOfficialAccountMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.msgKey)) {
            body["msgKey"] = request.msgKey;
        }
        if (!tea_util_1.default.isUnset(request.msgParam)) {
            body["msgParam"] = request.msgParam;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushOfficialAccountMessage",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/officialAccounts/robots/oToMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushOfficialAccountMessageResponse({}));
    }
    async pushOfficialAccountMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushOfficialAccountMessageHeaders({});
        return await this.pushOfficialAccountMessageWithOptions(request, headers, runtime);
    }
    async pushRobotMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.chatbotId)) {
            body["chatbotId"] = request.chatbotId;
        }
        if (!tea_util_1.default.isUnset(request.msgKey)) {
            body["msgKey"] = request.msgKey;
        }
        if (!tea_util_1.default.isUnset(request.msgParam)) {
            body["msgParam"] = request.msgParam;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushRobotMessage",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/robots/oToMessages/send`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushRobotMessageResponse({}));
    }
    async pushRobotMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushRobotMessageHeaders({});
        return await this.pushRobotMessageWithOptions(request, headers, runtime);
    }
    async replyRobotWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.proxyMessageStr)) {
            body["proxyMessageStr"] = request.proxyMessageStr;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "ReplyRobot",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/robots/reply`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "json",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new ReplyRobotResponse({}));
    }
    async replyRobot(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ReplyRobotHeaders({});
        return await this.replyRobotWithOptions(request, headers, runtime);
    }
    async updateOfficialAccountRobotInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.type)) {
            query["type"] = request.type;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.avatar)) {
            body["avatar"] = request.avatar;
        }
        if (!tea_util_1.default.isUnset(request.brief)) {
            body["brief"] = request.brief;
        }
        if (!tea_util_1.default.isUnset(request.description)) {
            body["description"] = request.description;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.previewMediaUrl)) {
            body["previewMediaUrl"] = request.previewMediaUrl;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "UpdateOfficialAccountRobotInfo",
            version: "dingmi_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/dingmi/officialAccounts/robots`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpdateOfficialAccountRobotInfoResponse({}));
    }
    async updateOfficialAccountRobotInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateOfficialAccountRobotInfoHeaders({});
        return await this.updateOfficialAccountRobotInfoWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
