// @alicloud/dingtalk/dist/bay_max_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class QueryBaymaxSkillLogHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryBaymaxSkillLogHeaders = QueryBaymaxSkillLogHeaders;
class QueryBaymaxSkillLogRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            from: 'from',
            skillExecuteId: 'skillExecuteId',
            to: 'to',
        };
    }
    static types() {
        return {
            from: 'number',
            skillExecuteId: 'string',
            to: 'number',
        };
    }
}
exports.QueryBaymaxSkillLogRequest = QueryBaymaxSkillLogRequest;
class QueryBaymaxSkillLogResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.QueryBaymaxSkillLogResponseBody = QueryBaymaxSkillLogResponseBody;
class QueryBaymaxSkillLogResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryBaymaxSkillLogResponseBody,
        };
    }
}
exports.QueryBaymaxSkillLogResponse = QueryBaymaxSkillLogResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async queryBaymaxSkillLogWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.from)) {
            query["from"] = request.from;
        }
        if (!tea_util_1.default.isUnset(request.skillExecuteId)) {
            query["skillExecuteId"] = request.skillExecuteId;
        }
        if (!tea_util_1.default.isUnset(request.to)) {
            query["to"] = request.to;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryBaymaxSkillLog",
            version: "bayMax_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/bayMax/skills/logs`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryBaymaxSkillLogResponse({}));
    }
    async queryBaymaxSkillLog(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryBaymaxSkillLogHeaders({});
        return await this.queryBaymaxSkillLogWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
