// @alicloud/dingtalk/dist/pedia_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class PediaWordsAddHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PediaWordsAddHeaders = PediaWordsAddHeaders;
class PediaWordsAddRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            contactList: 'contactList',
            highLightWordAlias: 'highLightWordAlias',
            picList: 'picList',
            relatedDoc: 'relatedDoc',
            relatedLink: 'relatedLink',
            userId: 'userId',
            wordAlias: 'wordAlias',
            wordName: 'wordName',
            wordParaphrase: 'wordParaphrase',
        };
    }
    static types() {
        return {
            contactList: { 'type': 'array', 'itemType': PediaWordsAddRequestContactList },
            highLightWordAlias: { 'type': 'array', 'itemType': 'string' },
            picList: { 'type': 'array', 'itemType': PediaWordsAddRequestPicList },
            relatedDoc: { 'type': 'array', 'itemType': PediaWordsAddRequestRelatedDoc },
            relatedLink: { 'type': 'array', 'itemType': PediaWordsAddRequestRelatedLink },
            userId: 'string',
            wordAlias: { 'type': 'array', 'itemType': 'string' },
            wordName: 'string',
            wordParaphrase: 'string',
        };
    }
}
exports.PediaWordsAddRequest = PediaWordsAddRequest;
class PediaWordsAddResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            success: 'boolean',
            uuid: 'number',
        };
    }
}
exports.PediaWordsAddResponseBody = PediaWordsAddResponseBody;
class PediaWordsAddResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PediaWordsAddResponseBody,
        };
    }
}
exports.PediaWordsAddResponse = PediaWordsAddResponse;
class PediaWordsApproveHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PediaWordsApproveHeaders = PediaWordsApproveHeaders;
class PediaWordsApproveRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            approveReason: 'approveReason',
            approveStatus: 'approveStatus',
            imHighLight: 'imHighLight',
            simHighLight: 'simHighLight',
            userId: 'userId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            approveReason: 'string',
            approveStatus: 'string',
            imHighLight: 'boolean',
            simHighLight: 'boolean',
            userId: 'string',
            uuid: 'number',
        };
    }
}
exports.PediaWordsApproveRequest = PediaWordsApproveRequest;
class PediaWordsApproveResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.PediaWordsApproveResponseBody = PediaWordsApproveResponseBody;
class PediaWordsApproveResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PediaWordsApproveResponseBody,
        };
    }
}
exports.PediaWordsApproveResponse = PediaWordsApproveResponse;
class PediaWordsDeleteHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PediaWordsDeleteHeaders = PediaWordsDeleteHeaders;
class PediaWordsDeleteRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userId: 'userId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            userId: 'string',
            uuid: 'number',
        };
    }
}
exports.PediaWordsDeleteRequest = PediaWordsDeleteRequest;
class PediaWordsDeleteResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            success: 'boolean',
            uuid: 'number',
        };
    }
}
exports.PediaWordsDeleteResponseBody = PediaWordsDeleteResponseBody;
class PediaWordsDeleteResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PediaWordsDeleteResponseBody,
        };
    }
}
exports.PediaWordsDeleteResponse = PediaWordsDeleteResponse;
class PediaWordsQueryHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PediaWordsQueryHeaders = PediaWordsQueryHeaders;
class PediaWordsQueryRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userId: 'userId',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            userId: 'string',
            uuid: 'number',
        };
    }
}
exports.PediaWordsQueryRequest = PediaWordsQueryRequest;
class PediaWordsQueryResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            data: 'data',
            success: 'success',
        };
    }
    static types() {
        return {
            data: PediaWordsQueryResponseBodyData,
            success: 'boolean',
        };
    }
}
exports.PediaWordsQueryResponseBody = PediaWordsQueryResponseBody;
class PediaWordsQueryResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PediaWordsQueryResponseBody,
        };
    }
}
exports.PediaWordsQueryResponse = PediaWordsQueryResponse;
class PediaWordsSearchHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PediaWordsSearchHeaders = PediaWordsSearchHeaders;
class PediaWordsSearchRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            status: 'status',
            userId: 'userId',
            wordName: 'wordName',
        };
    }
    static types() {
        return {
            pageNumber: 'number',
            pageSize: 'number',
            status: 'string',
            userId: 'string',
            wordName: 'string',
        };
    }
}
exports.PediaWordsSearchRequest = PediaWordsSearchRequest;
class PediaWordsSearchResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            data: 'data',
            success: 'success',
        };
    }
    static types() {
        return {
            data: { 'type': 'array', 'itemType': PediaWordsSearchResponseBodyData },
            success: 'boolean',
        };
    }
}
exports.PediaWordsSearchResponseBody = PediaWordsSearchResponseBody;
class PediaWordsSearchResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PediaWordsSearchResponseBody,
        };
    }
}
exports.PediaWordsSearchResponse = PediaWordsSearchResponse;
class PediaWordsUpdateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PediaWordsUpdateHeaders = PediaWordsUpdateHeaders;
class PediaWordsUpdateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appLink: 'appLink',
            contactList: 'contactList',
            highLightWordAlias: 'highLightWordAlias',
            picList: 'picList',
            relatedDoc: 'relatedDoc',
            relatedLink: 'relatedLink',
            userId: 'userId',
            uuid: 'uuid',
            wordAlias: 'wordAlias',
            wordName: 'wordName',
            wordParaphrase: 'wordParaphrase',
        };
    }
    static types() {
        return {
            appLink: { 'type': 'array', 'itemType': PediaWordsUpdateRequestAppLink },
            contactList: { 'type': 'array', 'itemType': PediaWordsUpdateRequestContactList },
            highLightWordAlias: { 'type': 'array', 'itemType': 'string' },
            picList: { 'type': 'array', 'itemType': PediaWordsUpdateRequestPicList },
            relatedDoc: { 'type': 'array', 'itemType': PediaWordsUpdateRequestRelatedDoc },
            relatedLink: { 'type': 'array', 'itemType': PediaWordsUpdateRequestRelatedLink },
            userId: 'string',
            uuid: 'number',
            wordAlias: { 'type': 'array', 'itemType': 'string' },
            wordName: 'string',
            wordParaphrase: 'string',
        };
    }
}
exports.PediaWordsUpdateRequest = PediaWordsUpdateRequest;
class PediaWordsUpdateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
            uuid: 'uuid',
        };
    }
    static types() {
        return {
            success: 'boolean',
            uuid: 'number',
        };
    }
}
exports.PediaWordsUpdateResponseBody = PediaWordsUpdateResponseBody;
class PediaWordsUpdateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PediaWordsUpdateResponseBody,
        };
    }
}
exports.PediaWordsUpdateResponse = PediaWordsUpdateResponse;
class PediaWordsAddRequestContactList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarMediaId: 'avatarMediaId',
            nickName: 'nickName',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatarMediaId: 'string',
            nickName: 'string',
            userId: 'string',
        };
    }
}
exports.PediaWordsAddRequestContactList = PediaWordsAddRequestContactList;
class PediaWordsAddRequestPicList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mediaIdUrl: 'mediaIdUrl',
        };
    }
    static types() {
        return {
            mediaIdUrl: 'string',
        };
    }
}
exports.PediaWordsAddRequestPicList = PediaWordsAddRequestPicList;
class PediaWordsAddRequestRelatedDoc extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.PediaWordsAddRequestRelatedDoc = PediaWordsAddRequestRelatedDoc;
class PediaWordsAddRequestRelatedLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
        };
    }
}
exports.PediaWordsAddRequestRelatedLink = PediaWordsAddRequestRelatedLink;
class PediaWordsQueryResponseBodyDataAppLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appName: 'appName',
            iconLink: 'iconLink',
            pcLink: 'pcLink',
            phoneLink: 'phoneLink',
        };
    }
    static types() {
        return {
            appName: 'string',
            iconLink: 'string',
            pcLink: 'string',
            phoneLink: 'string',
        };
    }
}
exports.PediaWordsQueryResponseBodyDataAppLink = PediaWordsQueryResponseBodyDataAppLink;
class PediaWordsQueryResponseBodyDataContactList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarMediaId: 'avatarMediaId',
            nickName: 'nickName',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatarMediaId: 'string',
            nickName: 'string',
            userId: 'string',
        };
    }
}
exports.PediaWordsQueryResponseBodyDataContactList = PediaWordsQueryResponseBodyDataContactList;
class PediaWordsQueryResponseBodyDataPicList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mediaIdUrl: 'mediaIdUrl',
        };
    }
    static types() {
        return {
            mediaIdUrl: 'string',
        };
    }
}
exports.PediaWordsQueryResponseBodyDataPicList = PediaWordsQueryResponseBodyDataPicList;
class PediaWordsQueryResponseBodyDataRelatedDoc extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.PediaWordsQueryResponseBodyDataRelatedDoc = PediaWordsQueryResponseBodyDataRelatedDoc;
class PediaWordsQueryResponseBodyDataRelatedLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
        };
    }
}
exports.PediaWordsQueryResponseBodyDataRelatedLink = PediaWordsQueryResponseBodyDataRelatedLink;
class PediaWordsQueryResponseBodyData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appLink: 'appLink',
            approveName: 'approveName',
            contactList: 'contactList',
            contacts: 'contacts',
            creatorName: 'creatorName',
            gmtCreate: 'gmtCreate',
            gmtModify: 'gmtModify',
            highLightWordAlias: 'highLightWordAlias',
            imHighLight: 'imHighLight',
            parentUuid: 'parentUuid',
            picList: 'picList',
            relatedDoc: 'relatedDoc',
            relatedLink: 'relatedLink',
            simHighLight: 'simHighLight',
            simpleWordParaphrase: 'simpleWordParaphrase',
            tagsList: 'tagsList',
            updaterName: 'updaterName',
            userId: 'userId',
            uuid: 'uuid',
            wordAlias: 'wordAlias',
            wordName: 'wordName',
            wordParaphrase: 'wordParaphrase',
        };
    }
    static types() {
        return {
            appLink: { 'type': 'array', 'itemType': PediaWordsQueryResponseBodyDataAppLink },
            approveName: 'string',
            contactList: { 'type': 'array', 'itemType': PediaWordsQueryResponseBodyDataContactList },
            contacts: { 'type': 'array', 'itemType': 'string' },
            creatorName: 'string',
            gmtCreate: 'number',
            gmtModify: 'number',
            highLightWordAlias: { 'type': 'array', 'itemType': 'string' },
            imHighLight: 'boolean',
            parentUuid: 'number',
            picList: { 'type': 'array', 'itemType': PediaWordsQueryResponseBodyDataPicList },
            relatedDoc: { 'type': 'array', 'itemType': PediaWordsQueryResponseBodyDataRelatedDoc },
            relatedLink: { 'type': 'array', 'itemType': PediaWordsQueryResponseBodyDataRelatedLink },
            simHighLight: 'boolean',
            simpleWordParaphrase: 'string',
            tagsList: { 'type': 'array', 'itemType': 'string' },
            updaterName: 'string',
            userId: 'string',
            uuid: 'number',
            wordAlias: { 'type': 'array', 'itemType': 'string' },
            wordName: 'string',
            wordParaphrase: 'string',
        };
    }
}
exports.PediaWordsQueryResponseBodyData = PediaWordsQueryResponseBodyData;
class PediaWordsSearchResponseBodyDataAppLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appName: 'appName',
            iconLink: 'iconLink',
            pcLink: 'pcLink',
            phoneLink: 'phoneLink',
        };
    }
    static types() {
        return {
            appName: 'string',
            iconLink: 'string',
            pcLink: 'string',
            phoneLink: 'string',
        };
    }
}
exports.PediaWordsSearchResponseBodyDataAppLink = PediaWordsSearchResponseBodyDataAppLink;
class PediaWordsSearchResponseBodyDataContactList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarMediaId: 'avatarMediaId',
            nickName: 'nickName',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatarMediaId: 'string',
            nickName: 'string',
            userId: 'string',
        };
    }
}
exports.PediaWordsSearchResponseBodyDataContactList = PediaWordsSearchResponseBodyDataContactList;
class PediaWordsSearchResponseBodyDataPicList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mediaIdUrl: 'mediaIdUrl',
        };
    }
    static types() {
        return {
            mediaIdUrl: 'string',
        };
    }
}
exports.PediaWordsSearchResponseBodyDataPicList = PediaWordsSearchResponseBodyDataPicList;
class PediaWordsSearchResponseBodyDataRelatedDoc extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.PediaWordsSearchResponseBodyDataRelatedDoc = PediaWordsSearchResponseBodyDataRelatedDoc;
class PediaWordsSearchResponseBodyDataRelatedLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.PediaWordsSearchResponseBodyDataRelatedLink = PediaWordsSearchResponseBodyDataRelatedLink;
class PediaWordsSearchResponseBodyData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appLink: 'appLink',
            approveName: 'approveName',
            contactList: 'contactList',
            contacts: 'contacts',
            creatorName: 'creatorName',
            gmtCreate: 'gmtCreate',
            gmtModify: 'gmtModify',
            highLightWordAlias: 'highLightWordAlias',
            imHighLight: 'imHighLight',
            parentUuid: 'parentUuid',
            picList: 'picList',
            relatedDoc: 'relatedDoc',
            relatedLink: 'relatedLink',
            simHighLight: 'simHighLight',
            simpleWordParaphrase: 'simpleWordParaphrase',
            tagsList: 'tagsList',
            updaterName: 'updaterName',
            userId: 'userId',
            uuid: 'uuid',
            wordAlias: 'wordAlias',
            wordName: 'wordName',
            wordParaphrase: 'wordParaphrase',
        };
    }
    static types() {
        return {
            appLink: { 'type': 'array', 'itemType': PediaWordsSearchResponseBodyDataAppLink },
            approveName: 'string',
            contactList: { 'type': 'array', 'itemType': PediaWordsSearchResponseBodyDataContactList },
            contacts: { 'type': 'array', 'itemType': 'string' },
            creatorName: 'string',
            gmtCreate: 'number',
            gmtModify: 'number',
            highLightWordAlias: { 'type': 'array', 'itemType': 'string' },
            imHighLight: 'boolean',
            parentUuid: 'number',
            picList: { 'type': 'array', 'itemType': PediaWordsSearchResponseBodyDataPicList },
            relatedDoc: { 'type': 'array', 'itemType': PediaWordsSearchResponseBodyDataRelatedDoc },
            relatedLink: { 'type': 'array', 'itemType': PediaWordsSearchResponseBodyDataRelatedLink },
            simHighLight: 'boolean',
            simpleWordParaphrase: 'string',
            tagsList: { 'type': 'array', 'itemType': 'string' },
            updaterName: 'string',
            userId: 'string',
            uuid: 'number',
            wordAlias: { 'type': 'array', 'itemType': 'string' },
            wordName: 'string',
            wordParaphrase: 'string',
        };
    }
}
exports.PediaWordsSearchResponseBodyData = PediaWordsSearchResponseBodyData;
class PediaWordsUpdateRequestAppLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appName: 'appName',
            iconLink: 'iconLink',
            pcLink: 'pcLink',
            phoneLink: 'phoneLink',
        };
    }
    static types() {
        return {
            appName: 'string',
            iconLink: 'string',
            pcLink: 'string',
            phoneLink: 'string',
        };
    }
}
exports.PediaWordsUpdateRequestAppLink = PediaWordsUpdateRequestAppLink;
class PediaWordsUpdateRequestContactList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarMediaId: 'avatarMediaId',
            nickName: 'nickName',
            userId: 'userId',
        };
    }
    static types() {
        return {
            avatarMediaId: 'string',
            nickName: 'string',
            userId: 'string',
        };
    }
}
exports.PediaWordsUpdateRequestContactList = PediaWordsUpdateRequestContactList;
class PediaWordsUpdateRequestPicList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mediaIdUrl: 'mediaIdUrl',
        };
    }
    static types() {
        return {
            mediaIdUrl: 'string',
        };
    }
}
exports.PediaWordsUpdateRequestPicList = PediaWordsUpdateRequestPicList;
class PediaWordsUpdateRequestRelatedDoc extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
            type: 'string',
        };
    }
}
exports.PediaWordsUpdateRequestRelatedDoc = PediaWordsUpdateRequestRelatedDoc;
class PediaWordsUpdateRequestRelatedLink extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            link: 'link',
            name: 'name',
        };
    }
    static types() {
        return {
            link: 'string',
            name: 'string',
        };
    }
}
exports.PediaWordsUpdateRequestRelatedLink = PediaWordsUpdateRequestRelatedLink;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async pediaWordsAddWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.contactList)) {
            body["contactList"] = request.contactList;
        }
        if (!tea_util_1.default.isUnset(request.highLightWordAlias)) {
            body["highLightWordAlias"] = request.highLightWordAlias;
        }
        if (!tea_util_1.default.isUnset(request.picList)) {
            body["picList"] = request.picList;
        }
        if (!tea_util_1.default.isUnset(request.relatedDoc)) {
            body["relatedDoc"] = request.relatedDoc;
        }
        if (!tea_util_1.default.isUnset(request.relatedLink)) {
            body["relatedLink"] = request.relatedLink;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.wordAlias)) {
            body["wordAlias"] = request.wordAlias;
        }
        if (!tea_util_1.default.isUnset(request.wordName)) {
            body["wordName"] = request.wordName;
        }
        if (!tea_util_1.default.isUnset(request.wordParaphrase)) {
            body["wordParaphrase"] = request.wordParaphrase;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PediaWordsAdd",
            version: "pedia_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/pedia/words`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PediaWordsAddResponse({}));
    }
    async pediaWordsAdd(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PediaWordsAddHeaders({});
        return await this.pediaWordsAddWithOptions(request, headers, runtime);
    }
    async pediaWordsApproveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.approveReason)) {
            body["approveReason"] = request.approveReason;
        }
        if (!tea_util_1.default.isUnset(request.approveStatus)) {
            body["approveStatus"] = request.approveStatus;
        }
        if (!tea_util_1.default.isUnset(request.imHighLight)) {
            body["imHighLight"] = request.imHighLight;
        }
        if (!tea_util_1.default.isUnset(request.simHighLight)) {
            body["simHighLight"] = request.simHighLight;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.uuid)) {
            body["uuid"] = request.uuid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PediaWordsApprove",
            version: "pedia_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/pedia/words/approve`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PediaWordsApproveResponse({}));
    }
    async pediaWordsApprove(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PediaWordsApproveHeaders({});
        return await this.pediaWordsApproveWithOptions(request, headers, runtime);
    }
    async pediaWordsDeleteWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.uuid)) {
            query["uuid"] = request.uuid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "PediaWordsDelete",
            version: "pedia_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/pedia/words`,
            method: "DELETE",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PediaWordsDeleteResponse({}));
    }
    async pediaWordsDelete(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PediaWordsDeleteHeaders({});
        return await this.pediaWordsDeleteWithOptions(request, headers, runtime);
    }
    async pediaWordsQueryWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.uuid)) {
            query["uuid"] = request.uuid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "PediaWordsQuery",
            version: "pedia_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/pedia/words/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PediaWordsQueryResponse({}));
    }
    async pediaWordsQuery(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PediaWordsQueryHeaders({});
        return await this.pediaWordsQueryWithOptions(request, headers, runtime);
    }
    async pediaWordsSearchWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            body["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            body["pageSize"] = request.pageSize;
        }
        if (!tea_util_1.default.isUnset(request.status)) {
            body["status"] = request.status;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.wordName)) {
            body["wordName"] = request.wordName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PediaWordsSearch",
            version: "pedia_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/pedia/words/search`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PediaWordsSearchResponse({}));
    }
    async pediaWordsSearch(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PediaWordsSearchHeaders({});
        return await this.pediaWordsSearchWithOptions(request, headers, runtime);
    }
    async pediaWordsUpdateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appLink)) {
            body["appLink"] = request.appLink;
        }
        if (!tea_util_1.default.isUnset(request.contactList)) {
            body["contactList"] = request.contactList;
        }
        if (!tea_util_1.default.isUnset(request.highLightWordAlias)) {
            body["highLightWordAlias"] = request.highLightWordAlias;
        }
        if (!tea_util_1.default.isUnset(request.picList)) {
            body["picList"] = request.picList;
        }
        if (!tea_util_1.default.isUnset(request.relatedDoc)) {
            body["relatedDoc"] = request.relatedDoc;
        }
        if (!tea_util_1.default.isUnset(request.relatedLink)) {
            body["relatedLink"] = request.relatedLink;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        if (!tea_util_1.default.isUnset(request.uuid)) {
            body["uuid"] = request.uuid;
        }
        if (!tea_util_1.default.isUnset(request.wordAlias)) {
            body["wordAlias"] = request.wordAlias;
        }
        if (!tea_util_1.default.isUnset(request.wordName)) {
            body["wordName"] = request.wordName;
        }
        if (!tea_util_1.default.isUnset(request.wordParaphrase)) {
            body["wordParaphrase"] = request.wordParaphrase;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PediaWordsUpdate",
            version: "pedia_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/pedia/words`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PediaWordsUpdateResponse({}));
    }
    async pediaWordsUpdate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PediaWordsUpdateHeaders({});
        return await this.pediaWordsUpdateWithOptions(request, headers, runtime);
    }
}
exports.default = Client;
