// @alicloud/dingtalk/dist/diot_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const gateway_dingtalk_1 = __importDefault(require("@alicloud/gateway-dingtalk"));
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AyunOnlienTestRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            reqId: 'reqId',
        };
    }
    static types() {
        return {
            reqId: 'string',
        };
    }
}
exports.AyunOnlienTestRequest = AyunOnlienTestRequest;
class AyunOnlienTestResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.AyunOnlienTestResponseBody = AyunOnlienTestResponseBody;
class AyunOnlienTestResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: AyunOnlienTestResponseBody,
        };
    }
}
exports.AyunOnlienTestResponse = AyunOnlienTestResponse;
class BatchDeleteDeviceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BatchDeleteDeviceHeaders = BatchDeleteDeviceHeaders;
class BatchDeleteDeviceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            deviceIds: 'deviceIds',
        };
    }
    static types() {
        return {
            corpId: 'string',
            deviceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.BatchDeleteDeviceRequest = BatchDeleteDeviceRequest;
class BatchDeleteDeviceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceIds: 'deviceIds',
        };
    }
    static types() {
        return {
            deviceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.BatchDeleteDeviceResponseBody = BatchDeleteDeviceResponseBody;
class BatchDeleteDeviceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: BatchDeleteDeviceResponseBody,
        };
    }
}
exports.BatchDeleteDeviceResponse = BatchDeleteDeviceResponse;
class BatchRegisterDeviceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BatchRegisterDeviceHeaders = BatchRegisterDeviceHeaders;
class BatchRegisterDeviceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            devices: 'devices',
        };
    }
    static types() {
        return {
            corpId: 'string',
            devices: { 'type': 'array', 'itemType': BatchRegisterDeviceRequestDevices },
        };
    }
}
exports.BatchRegisterDeviceRequest = BatchRegisterDeviceRequest;
class BatchRegisterDeviceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceIds: 'deviceIds',
        };
    }
    static types() {
        return {
            deviceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.BatchRegisterDeviceResponseBody = BatchRegisterDeviceResponseBody;
class BatchRegisterDeviceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: BatchRegisterDeviceResponseBody,
        };
    }
}
exports.BatchRegisterDeviceResponse = BatchRegisterDeviceResponse;
class BatchRegisterEventTypeHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BatchRegisterEventTypeHeaders = BatchRegisterEventTypeHeaders;
class BatchRegisterEventTypeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            eventTypes: 'eventTypes',
        };
    }
    static types() {
        return {
            corpId: 'string',
            eventTypes: { 'type': 'array', 'itemType': BatchRegisterEventTypeRequestEventTypes },
        };
    }
}
exports.BatchRegisterEventTypeRequest = BatchRegisterEventTypeRequest;
class BatchRegisterEventTypeResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            eventTypes: 'eventTypes',
        };
    }
    static types() {
        return {
            eventTypes: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.BatchRegisterEventTypeResponseBody = BatchRegisterEventTypeResponseBody;
class BatchRegisterEventTypeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: BatchRegisterEventTypeResponseBody,
        };
    }
}
exports.BatchRegisterEventTypeResponse = BatchRegisterEventTypeResponse;
class BatchUpdateDeviceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BatchUpdateDeviceHeaders = BatchUpdateDeviceHeaders;
class BatchUpdateDeviceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            devices: 'devices',
        };
    }
    static types() {
        return {
            corpId: 'string',
            devices: { 'type': 'array', 'itemType': BatchUpdateDeviceRequestDevices },
        };
    }
}
exports.BatchUpdateDeviceRequest = BatchUpdateDeviceRequest;
class BatchUpdateDeviceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceIds: 'deviceIds',
        };
    }
    static types() {
        return {
            deviceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.BatchUpdateDeviceResponseBody = BatchUpdateDeviceResponseBody;
class BatchUpdateDeviceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: BatchUpdateDeviceResponseBody,
        };
    }
}
exports.BatchUpdateDeviceResponse = BatchUpdateDeviceResponse;
class BindSystemHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BindSystemHeaders = BindSystemHeaders;
class BindSystemRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authCode: 'authCode',
            clientId: 'clientId',
            clientName: 'clientName',
            corpId: 'corpId',
            extraData: 'extraData',
        };
    }
    static types() {
        return {
            authCode: 'string',
            clientId: 'string',
            clientName: 'string',
            corpId: 'string',
            extraData: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.BindSystemRequest = BindSystemRequest;
class BindSystemResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            clientId: 'clientId',
            corpId: 'corpId',
        };
    }
    static types() {
        return {
            clientId: 'string',
            corpId: 'string',
        };
    }
}
exports.BindSystemResponseBody = BindSystemResponseBody;
class BindSystemResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: BindSystemResponseBody,
        };
    }
}
exports.BindSystemResponse = BindSystemResponse;
class DeviceConferenceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DeviceConferenceHeaders = DeviceConferenceHeaders;
class DeviceConferenceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            confTitle: 'confTitle',
            conferenceId: 'conferenceId',
            conferencePassword: 'conferencePassword',
            deviceIds: 'deviceIds',
        };
    }
    static types() {
        return {
            confTitle: 'string',
            conferenceId: 'string',
            conferencePassword: 'string',
            deviceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.DeviceConferenceRequest = DeviceConferenceRequest;
class DeviceConferenceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            conferenceId: 'conferenceId',
        };
    }
    static types() {
        return {
            conferenceId: 'string',
        };
    }
}
exports.DeviceConferenceResponseBody = DeviceConferenceResponseBody;
class DeviceConferenceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DeviceConferenceResponseBody,
        };
    }
}
exports.DeviceConferenceResponse = DeviceConferenceResponse;
class DiotMarketManagerTestResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.DiotMarketManagerTestResponseBody = DiotMarketManagerTestResponseBody;
class DiotMarketManagerTestResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DiotMarketManagerTestResponseBody,
        };
    }
}
exports.DiotMarketManagerTestResponse = DiotMarketManagerTestResponse;
class DiotSystemMarkTestResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.DiotSystemMarkTestResponseBody = DiotSystemMarkTestResponseBody;
class DiotSystemMarkTestResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DiotSystemMarkTestResponseBody,
        };
    }
}
exports.DiotSystemMarkTestResponse = DiotSystemMarkTestResponse;
class DiotMarketManagerResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.DiotMarketManagerResponseBody = DiotMarketManagerResponseBody;
class DiotMarketManagerResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: DiotMarketManagerResponseBody,
        };
    }
}
exports.DiotMarketManagerResponse = DiotMarketManagerResponse;
class PushEventHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.PushEventHeaders = PushEventHeaders;
class PushEventRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            deviceId: 'deviceId',
            eventId: 'eventId',
            eventName: 'eventName',
            eventType: 'eventType',
            extraData: 'extraData',
            location: 'location',
            msg: 'msg',
            occurrenceTime: 'occurrenceTime',
            picUrls: 'picUrls',
        };
    }
    static types() {
        return {
            corpId: 'string',
            deviceId: 'string',
            eventId: 'string',
            eventName: 'string',
            eventType: 'string',
            extraData: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            location: 'string',
            msg: 'string',
            occurrenceTime: 'number',
            picUrls: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.PushEventRequest = PushEventRequest;
class PushEventResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            eventId: 'eventId',
        };
    }
    static types() {
        return {
            eventId: 'string',
        };
    }
}
exports.PushEventResponseBody = PushEventResponseBody;
class PushEventResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: PushEventResponseBody,
        };
    }
}
exports.PushEventResponse = PushEventResponse;
class QueryDeviceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryDeviceHeaders = QueryDeviceHeaders;
class QueryDeviceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
        };
    }
    static types() {
        return {
            corpId: 'string',
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
}
exports.QueryDeviceRequest = QueryDeviceRequest;
class QueryDeviceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            data: 'data',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            data: { 'type': 'array', 'itemType': QueryDeviceResponseBodyData },
            pageNumber: 'number',
            pageSize: 'number',
            totalCount: 'number',
        };
    }
}
exports.QueryDeviceResponseBody = QueryDeviceResponseBody;
class QueryDeviceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryDeviceResponseBody,
        };
    }
}
exports.QueryDeviceResponse = QueryDeviceResponse;
class QueryEventHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryEventHeaders = QueryEventHeaders;
class QueryEventRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            deviceIdList: 'deviceIdList',
            endTime: 'endTime',
            eventId: 'eventId',
            eventStatusList: 'eventStatusList',
            eventTypeList: 'eventTypeList',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            startTime: 'startTime',
        };
    }
    static types() {
        return {
            corpId: 'string',
            deviceIdList: { 'type': 'array', 'itemType': 'string' },
            endTime: 'number',
            eventId: 'string',
            eventStatusList: { 'type': 'array', 'itemType': 'number' },
            eventTypeList: { 'type': 'array', 'itemType': 'string' },
            pageNumber: 'number',
            pageSize: 'number',
            startTime: 'number',
        };
    }
}
exports.QueryEventRequest = QueryEventRequest;
class QueryEventResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            data: 'data',
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            data: { 'type': 'array', 'itemType': QueryEventResponseBodyData },
            pageNumber: 'number',
            pageSize: 'number',
            totalCount: 'number',
        };
    }
}
exports.QueryEventResponseBody = QueryEventResponseBody;
class QueryEventResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: QueryEventResponseBody,
        };
    }
}
exports.QueryEventResponse = QueryEventResponse;
class RegisterDeviceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.RegisterDeviceHeaders = RegisterDeviceHeaders;
class RegisterDeviceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            deviceName: 'deviceName',
            deviceStatus: 'deviceStatus',
            deviceType: 'deviceType',
            deviceTypeName: 'deviceTypeName',
            id: 'id',
            liveUrls: 'liveUrls',
            location: 'location',
            nickName: 'nickName',
            parentId: 'parentId',
            productType: 'productType',
        };
    }
    static types() {
        return {
            corpId: 'string',
            deviceName: 'string',
            deviceStatus: 'number',
            deviceType: 'string',
            deviceTypeName: 'string',
            id: 'string',
            liveUrls: RegisterDeviceRequestLiveUrls,
            location: 'string',
            nickName: 'string',
            parentId: 'string',
            productType: 'string',
        };
    }
}
exports.RegisterDeviceRequest = RegisterDeviceRequest;
class RegisterDeviceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
        };
    }
    static types() {
        return {
            deviceId: 'string',
        };
    }
}
exports.RegisterDeviceResponseBody = RegisterDeviceResponseBody;
class RegisterDeviceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: RegisterDeviceResponseBody,
        };
    }
}
exports.RegisterDeviceResponse = RegisterDeviceResponse;
class UpgradeDeviceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.UpgradeDeviceResponseBody = UpgradeDeviceResponseBody;
class UpgradeDeviceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: UpgradeDeviceResponseBody,
        };
    }
}
exports.UpgradeDeviceResponse = UpgradeDeviceResponse;
class WorkbenchTransformInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.WorkbenchTransformInfoResponseBody = WorkbenchTransformInfoResponseBody;
class WorkbenchTransformInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            statusCode: 'statusCode',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            statusCode: 'number',
            body: WorkbenchTransformInfoResponseBody,
        };
    }
}
exports.WorkbenchTransformInfoResponse = WorkbenchTransformInfoResponse;
class BatchRegisterDeviceRequestDevicesLiveUrls extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            flv: 'flv',
            hls: 'hls',
            rtmp: 'rtmp',
        };
    }
    static types() {
        return {
            flv: 'string',
            hls: 'string',
            rtmp: 'string',
        };
    }
}
exports.BatchRegisterDeviceRequestDevicesLiveUrls = BatchRegisterDeviceRequestDevicesLiveUrls;
class BatchRegisterDeviceRequestDevices extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            deviceStatus: 'deviceStatus',
            deviceType: 'deviceType',
            deviceTypeName: 'deviceTypeName',
            extraData: 'extraData',
            liveUrls: 'liveUrls',
            location: 'location',
            parentId: 'parentId',
            productType: 'productType',
        };
    }
    static types() {
        return {
            deviceId: 'string',
            deviceName: 'string',
            deviceStatus: 'number',
            deviceType: 'string',
            deviceTypeName: 'string',
            extraData: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            liveUrls: BatchRegisterDeviceRequestDevicesLiveUrls,
            location: 'string',
            parentId: 'string',
            productType: 'string',
        };
    }
}
exports.BatchRegisterDeviceRequestDevices = BatchRegisterDeviceRequestDevices;
class BatchRegisterEventTypeRequestEventTypes extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            eventType: 'eventType',
            eventTypeName: 'eventTypeName',
        };
    }
    static types() {
        return {
            eventType: 'string',
            eventTypeName: 'string',
        };
    }
}
exports.BatchRegisterEventTypeRequestEventTypes = BatchRegisterEventTypeRequestEventTypes;
class BatchUpdateDeviceRequestDevicesLiveUrls extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            flv: 'flv',
            hls: 'hls',
            rtmp: 'rtmp',
        };
    }
    static types() {
        return {
            flv: 'string',
            hls: 'string',
            rtmp: 'string',
        };
    }
}
exports.BatchUpdateDeviceRequestDevicesLiveUrls = BatchUpdateDeviceRequestDevicesLiveUrls;
class BatchUpdateDeviceRequestDevices extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            deviceStatus: 'deviceStatus',
            extraData: 'extraData',
            liveUrls: 'liveUrls',
            location: 'location',
        };
    }
    static types() {
        return {
            deviceId: 'string',
            deviceName: 'string',
            deviceStatus: 'number',
            extraData: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            liveUrls: BatchUpdateDeviceRequestDevicesLiveUrls,
            location: 'string',
        };
    }
}
exports.BatchUpdateDeviceRequestDevices = BatchUpdateDeviceRequestDevices;
class QueryDeviceResponseBodyDataLiveUrls extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            flv: 'flv',
            hls: 'hls',
            rtmp: 'rtmp',
        };
    }
    static types() {
        return {
            flv: 'string',
            hls: 'string',
            rtmp: 'string',
        };
    }
}
exports.QueryDeviceResponseBodyDataLiveUrls = QueryDeviceResponseBodyDataLiveUrls;
class QueryDeviceResponseBodyData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            deviceStatus: 'deviceStatus',
            deviceType: 'deviceType',
            deviceTypeName: 'deviceTypeName',
            liveUrls: 'liveUrls',
            location: 'location',
            parentId: 'parentId',
            productType: 'productType',
        };
    }
    static types() {
        return {
            deviceId: 'string',
            deviceName: 'string',
            deviceStatus: 'number',
            deviceType: 'string',
            deviceTypeName: 'string',
            liveUrls: QueryDeviceResponseBodyDataLiveUrls,
            location: 'string',
            parentId: 'string',
            productType: 'string',
        };
    }
}
exports.QueryDeviceResponseBodyData = QueryDeviceResponseBodyData;
class QueryEventResponseBodyData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            eventId: 'eventId',
            eventName: 'eventName',
            eventStatus: 'eventStatus',
            eventType: 'eventType',
            msg: 'msg',
            occurrenceTime: 'occurrenceTime',
            picUrls: 'picUrls',
        };
    }
    static types() {
        return {
            eventId: 'string',
            eventName: 'string',
            eventStatus: 'number',
            eventType: 'string',
            msg: 'string',
            occurrenceTime: 'number',
            picUrls: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.QueryEventResponseBodyData = QueryEventResponseBodyData;
class RegisterDeviceRequestLiveUrls extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            flv: 'flv',
            hls: 'hls',
            rtmp: 'rtmp',
        };
    }
    static types() {
        return {
            flv: 'string',
            hls: 'string',
            rtmp: 'string',
        };
    }
}
exports.RegisterDeviceRequestLiveUrls = RegisterDeviceRequestLiveUrls;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._client = new gateway_dingtalk_1.default();
        this._spi = this._client;
        this._signatureAlgorithm = "v2";
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async ayunOnlienTestWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.reqId)) {
            query["reqId"] = request.reqId;
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "AyunOnlienTest",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/ayunTest`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new AyunOnlienTestResponse({}));
    }
    async ayunOnlienTest(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.ayunOnlienTestWithOptions(request, headers, runtime);
    }
    async batchDeleteDeviceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.deviceIds)) {
            body["deviceIds"] = request.deviceIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "BatchDeleteDevice",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/devices/remove`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new BatchDeleteDeviceResponse({}));
    }
    async batchDeleteDevice(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BatchDeleteDeviceHeaders({});
        return await this.batchDeleteDeviceWithOptions(request, headers, runtime);
    }
    async batchRegisterDeviceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.devices)) {
            body["devices"] = request.devices;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "BatchRegisterDevice",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/devices/registrations/batch`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new BatchRegisterDeviceResponse({}));
    }
    async batchRegisterDevice(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BatchRegisterDeviceHeaders({});
        return await this.batchRegisterDeviceWithOptions(request, headers, runtime);
    }
    async batchRegisterEventTypeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.eventTypes)) {
            body["eventTypes"] = request.eventTypes;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "BatchRegisterEventType",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/eventTypes/registrations/batch`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new BatchRegisterEventTypeResponse({}));
    }
    async batchRegisterEventType(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BatchRegisterEventTypeHeaders({});
        return await this.batchRegisterEventTypeWithOptions(request, headers, runtime);
    }
    async batchUpdateDeviceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.devices)) {
            body["devices"] = request.devices;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "BatchUpdateDevice",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/devices/batch`,
            method: "PUT",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new BatchUpdateDeviceResponse({}));
    }
    async batchUpdateDevice(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BatchUpdateDeviceHeaders({});
        return await this.batchUpdateDeviceWithOptions(request, headers, runtime);
    }
    async bindSystemWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.authCode)) {
            body["authCode"] = request.authCode;
        }
        if (!tea_util_1.default.isUnset(request.clientId)) {
            body["clientId"] = request.clientId;
        }
        if (!tea_util_1.default.isUnset(request.clientName)) {
            body["clientName"] = request.clientName;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.extraData)) {
            body["extraData"] = request.extraData;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "BindSystem",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/systems/bind`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new BindSystemResponse({}));
    }
    async bindSystem(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BindSystemHeaders({});
        return await this.bindSystemWithOptions(request, headers, runtime);
    }
    async deviceConferenceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.confTitle)) {
            body["confTitle"] = request.confTitle;
        }
        if (!tea_util_1.default.isUnset(request.conferenceId)) {
            body["conferenceId"] = request.conferenceId;
        }
        if (!tea_util_1.default.isUnset(request.conferencePassword)) {
            body["conferencePassword"] = request.conferencePassword;
        }
        if (!tea_util_1.default.isUnset(request.deviceIds)) {
            body["deviceIds"] = request.deviceIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "DeviceConference",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/deviceConferences/initiate`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DeviceConferenceResponse({}));
    }
    async deviceConference(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeviceConferenceHeaders({});
        return await this.deviceConferenceWithOptions(request, headers, runtime);
    }
    async diotMarketManagerTestWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "DiotMarketManagerTest",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/market/manager/test`,
            method: "PUT",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DiotMarketManagerTestResponse({}));
    }
    async diotMarketManagerTest() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.diotMarketManagerTestWithOptions(headers, runtime);
    }
    async diotSystemMarkTestWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "DiotSystemMarkTest",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/sys/mark/test`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DiotSystemMarkTestResponse({}));
    }
    async diotSystemMarkTest() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.diotSystemMarkTestWithOptions(headers, runtime);
    }
    async diot_Market_ManagerWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "Diot_Market_Manager",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/market/manager`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new DiotMarketManagerResponse({}));
    }
    async diot_Market_Manager() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.diot_Market_ManagerWithOptions(headers, runtime);
    }
    async pushEventWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.deviceId)) {
            body["deviceId"] = request.deviceId;
        }
        if (!tea_util_1.default.isUnset(request.eventId)) {
            body["eventId"] = request.eventId;
        }
        if (!tea_util_1.default.isUnset(request.eventName)) {
            body["eventName"] = request.eventName;
        }
        if (!tea_util_1.default.isUnset(request.eventType)) {
            body["eventType"] = request.eventType;
        }
        if (!tea_util_1.default.isUnset(request.extraData)) {
            body["extraData"] = request.extraData;
        }
        if (!tea_util_1.default.isUnset(request.location)) {
            body["location"] = request.location;
        }
        if (!tea_util_1.default.isUnset(request.msg)) {
            body["msg"] = request.msg;
        }
        if (!tea_util_1.default.isUnset(request.occurrenceTime)) {
            body["occurrenceTime"] = request.occurrenceTime;
        }
        if (!tea_util_1.default.isUnset(request.picUrls)) {
            body["picUrls"] = request.picUrls;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "PushEvent",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/events/push`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new PushEventResponse({}));
    }
    async pushEvent(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new PushEventHeaders({});
        return await this.pushEventWithOptions(request, headers, runtime);
    }
    async queryDeviceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            query["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        let params = new $OpenApi.Params({
            action: "QueryDevice",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/devices`,
            method: "GET",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryDeviceResponse({}));
    }
    async queryDevice(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryDeviceHeaders({});
        return await this.queryDeviceWithOptions(request, headers, runtime);
    }
    async queryEventWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.deviceIdList)) {
            body["deviceIdList"] = request.deviceIdList;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.eventId)) {
            body["eventId"] = request.eventId;
        }
        if (!tea_util_1.default.isUnset(request.eventStatusList)) {
            body["eventStatusList"] = request.eventStatusList;
        }
        if (!tea_util_1.default.isUnset(request.eventTypeList)) {
            body["eventTypeList"] = request.eventTypeList;
        }
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            body["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            body["pageSize"] = request.pageSize;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "QueryEvent",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/events/query`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new QueryEventResponse({}));
    }
    async queryEvent(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryEventHeaders({});
        return await this.queryEventWithOptions(request, headers, runtime);
    }
    async registerDeviceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.deviceName)) {
            body["deviceName"] = request.deviceName;
        }
        if (!tea_util_1.default.isUnset(request.deviceStatus)) {
            body["deviceStatus"] = request.deviceStatus;
        }
        if (!tea_util_1.default.isUnset(request.deviceType)) {
            body["deviceType"] = request.deviceType;
        }
        if (!tea_util_1.default.isUnset(request.deviceTypeName)) {
            body["deviceTypeName"] = request.deviceTypeName;
        }
        if (!tea_util_1.default.isUnset(request.id)) {
            body["id"] = request.id;
        }
        if (!tea_util_1.default.isUnset(request.liveUrls)) {
            body["liveUrls"] = request.liveUrls;
        }
        if (!tea_util_1.default.isUnset(request.location)) {
            body["location"] = request.location;
        }
        if (!tea_util_1.default.isUnset(request.nickName)) {
            body["nickName"] = request.nickName;
        }
        if (!tea_util_1.default.isUnset(request.parentId)) {
            body["parentId"] = request.parentId;
        }
        if (!tea_util_1.default.isUnset(request.productType)) {
            body["productType"] = request.productType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        let params = new $OpenApi.Params({
            action: "RegisterDevice",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/devices/register`,
            method: "POST",
            authType: "AK",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new RegisterDeviceResponse({}));
    }
    async registerDevice(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RegisterDeviceHeaders({});
        return await this.registerDeviceWithOptions(request, headers, runtime);
    }
    async upgradeDeviceWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "UpgradeDevice",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/upgrade/device`,
            method: "POST",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new UpgradeDeviceResponse({}));
    }
    async upgradeDevice() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.upgradeDeviceWithOptions(headers, runtime);
    }
    async workbenchTransformInfoWithOptions(headers, runtime) {
        let req = new $OpenApi.OpenApiRequest({
            headers: headers,
        });
        let params = new $OpenApi.Params({
            action: "WorkbenchTransformInfo",
            version: "diot_1.0",
            protocol: "HTTP",
            pathname: `/v1.0/diot/workbench/transform`,
            method: "GET",
            authType: "Anonymous",
            style: "ROA",
            reqBodyType: "none",
            bodyType: "json",
        });
        return $tea.cast(await this.execute(params, req, runtime), new WorkbenchTransformInfoResponse({}));
    }
    async workbenchTransformInfo() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = {};
        return await this.workbenchTransformInfoWithOptions(headers, runtime);
    }
}
exports.default = Client;
